<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Briana Mittleman" />


<title>Nuclear fraction and nascent transcription</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/journal.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.0.13/css/fa-svg-with-js.css" rel="stylesheet" />
<script src="site_libs/font-awesome-5.0.13/js/fontawesome-all.min.js"></script>
<script src="site_libs/font-awesome-5.0.13/js/fa-v4-shims.min.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">apaQTL</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https/github.com/brimittleman/apaQTL">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Nuclear fraction and nascent transcription</h1>
<h4 class="author"><em>Briana Mittleman</em></h4>
<h4 class="date"><em>5/15/2019</em></h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2019-06-24
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>apaQTL/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.3.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity itâ€™s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20190411code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20190411)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20190411code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20190411)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g.Â subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcombrimittlemanapaQTLtree438b5c0afe6010686cad87f7c91599966dd48cc6targetblank438b5c0a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/brimittleman/apaQTL/tree/438b5c0afe6010686cad87f7c91599966dd48cc6" target="_blank">438b5c0</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcombrimittlemanapaQTLtree438b5c0afe6010686cad87f7c91599966dd48cc6targetblank438b5c0a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/.DS_Store
    Ignored:    output/.DS_Store

Untracked files:
    Untracked:  .Rprofile
    Untracked:  ._.DS_Store
    Untracked:  .gitignore
    Untracked:  _workflowr.yml
    Untracked:  analysis/._PASdescriptiveplots.Rmd
    Untracked:  analysis/._cuttoffPercUsage.Rmd
    Untracked:  analysis/QTLexampleplots.Rmd
    Untracked:  analysis/cuttoffPercUsage.Rmd
    Untracked:  analysis/eQTLoverlap.Rmd
    Untracked:  analysis/mergeRNA.Rmd
    Untracked:  analysis/oldstuffNotNeeded.Rmd
    Untracked:  apaQTL.Rproj
    Untracked:  code/.NascentRNAdtPlotFirstintronicPAS.sh.swp
    Untracked:  code/._ApaQTL_nominalNonnorm.sh
    Untracked:  code/._BothFracDTPlotGeneRegions_normalized.sh
    Untracked:  code/._EandPqtls.sh
    Untracked:  code/._FC_NucintornUpandDown.sh
    Untracked:  code/._FC_UTR.sh
    Untracked:  code/._FC_intornUpandDownsteamPAS.sh
    Untracked:  code/._FC_nascentseq.sh
    Untracked:  code/._FC_newPeaks_olddata.sh
    Untracked:  code/._HMMpermuteTotal.py
    Untracked:  code/._HmmPermute.py
    Untracked:  code/._LC_samplegroups.py
    Untracked:  code/._NascentRNAdtPlot.sh
    Untracked:  code/._NascentRNAdtPlot3UTRPAS.sh
    Untracked:  code/._NascentRNAdtPlotExcludeFirstintronicPAS.sh
    Untracked:  code/._NascentRNAdtPlotNucPAS.sh
    Untracked:  code/._NascentRNAdtPlotTotPAS.sh
    Untracked:  code/._NascentRNAdtPlotintronicPAS.sh
    Untracked:  code/._NascnetRNAdtPlotPAS.sh
    Untracked:  code/._NetSeq_fourthintronDT.sh
    Untracked:  code/._QTL2bed.py
    Untracked:  code/._QTL2bed_withstrand.py
    Untracked:  code/._RNAbam2bw.sh
    Untracked:  code/._SnakefilePAS
    Untracked:  code/._SnakefilefiltPAS
    Untracked:  code/._TESplots100bp.sh
    Untracked:  code/._TESplots150bp.sh
    Untracked:  code/._TESplots200bp.sh
    Untracked:  code/._Untitled
    Untracked:  code/._ZipandTabPheno.sh
    Untracked:  code/._aAPAqtl_nominal39ind.sh
    Untracked:  code/._apaQTLCorrectPvalMakeQQ.R
    Untracked:  code/._apaQTL_Nominal.sh
    Untracked:  code/._apaQTL_permuted.sh
    Untracked:  code/._assignNucIntonpeak2intronlocs.sh
    Untracked:  code/._assignTotIntronpeak2intronlocs.sh
    Untracked:  code/._bam2BW_5primemost.sh
    Untracked:  code/._bed2saf.py
    Untracked:  code/._bothFracDTplot1stintron.sh
    Untracked:  code/._bothFracDTplot4thintron.sh
    Untracked:  code/._bothFrac_FC.sh
    Untracked:  code/._callPeaksYL.py
    Untracked:  code/._changenomQTLres2geneName.py
    Untracked:  code/._chooseAnno2SAF.py
    Untracked:  code/._chooseSignalSite
    Untracked:  code/._chooseSignalSite.py
    Untracked:  code/._cluster.json
    Untracked:  code/._clusterPAS.json
    Untracked:  code/._clusterfiltPAS.json
    Untracked:  code/._codingdms2bed.py
    Untracked:  code/._config.yaml
    Untracked:  code/._config2.yaml
    Untracked:  code/._configOLD.yaml
    Untracked:  code/._convertNominal2SNPLOC.py
    Untracked:  code/._convertNumeric.py
    Untracked:  code/._correctNomeqtl.R
    Untracked:  code/._dag.pdf
    Untracked:  code/._eQTLgenestestedapa.py
    Untracked:  code/._encodeRNADTplots.sh
    Untracked:  code/._extractGenotypes.py
    Untracked:  code/._extractseqfromqtlfastq.py
    Untracked:  code/._fc2leafphen.py
    Untracked:  code/._filter5perc.R
    Untracked:  code/._filter5percPheno.py
    Untracked:  code/._filterpeaks.py
    Untracked:  code/._finalPASbed2SAF.py
    Untracked:  code/._fix4su304corr.py
    Untracked:  code/._fix4su604corr.py
    Untracked:  code/._fix4sukalisto.py
    Untracked:  code/._fixExandUnexeQTL
    Untracked:  code/._fixExandUnexeQTL.py
    Untracked:  code/._fixFChead.py
    Untracked:  code/._fixFChead_bothfrac.py
    Untracked:  code/._fixH3k12ac.py
    Untracked:  code/._fixRNAhead4corr.py
    Untracked:  code/._fixRNAkalisto.py
    Untracked:  code/._fixgroupedtranscript.py
    Untracked:  code/._fixhead_netseqfc.py
    Untracked:  code/._getAPAfromanyeQTL.py
    Untracked:  code/._getApapval4eqtl.py
    Untracked:  code/._getApapval4eqtl_unexp.py
    Untracked:  code/._getDownstreamIntronNuclear.py
    Untracked:  code/._getIntronDownstreamPAS.py
    Untracked:  code/._getIntronUpstreamPAS.py
    Untracked:  code/._getQTLalleles.py
    Untracked:  code/._getQTLfastq.sh
    Untracked:  code/._getUpstreamIntronNuclear.py
    Untracked:  code/._grouptranscripts.py
    Untracked:  code/._keep5perMAF.py
    Untracked:  code/._keepSNP_vcf.sh
    Untracked:  code/._make5percPeakbed.py
    Untracked:  code/._makeFileID.py
    Untracked:  code/._makePheno.py
    Untracked:  code/._makeSAFbothfrac5perc.py
    Untracked:  code/._makeSNP2rsidfile.py
    Untracked:  code/._makeeQTLempirical_unexp.py
    Untracked:  code/._makeeQTLempiricaldist.py
    Untracked:  code/._makegencondeTSSfile.py
    Untracked:  code/._mergRNABam.sh
    Untracked:  code/._mergeAllBam.sh
    Untracked:  code/._mergeBW_norm.sh
    Untracked:  code/._mergeBamNascent.sh
    Untracked:  code/._mergeByFracBam.sh
    Untracked:  code/._mergePeaks.sh
    Untracked:  code/._mnase1stintron.sh
    Untracked:  code/._mnaseDT_fourthintron.sh
    Untracked:  code/._namePeaks.py
    Untracked:  code/._netseqDTplot1stIntron.sh
    Untracked:  code/._netseqFC.sh
    Untracked:  code/._peak2PAS.py
    Untracked:  code/._peakFC.sh
    Untracked:  code/._pheno2countonly.R
    Untracked:  code/._phenoQTLfromlist.py
    Untracked:  code/._processYRIgen.py
    Untracked:  code/._qtlRegionseq.sh
    Untracked:  code/._qtlsPvalOppFrac.py
    Untracked:  code/._quantassign2parsedpeak.py
    Untracked:  code/._removeXfromHmm.py
    Untracked:  code/._removeloc_pheno.py
    Untracked:  code/._runCorrectNomEqtl.sh
    Untracked:  code/._runHMMpermuteAPAqtls.sh
    Untracked:  code/._runHMMpermuteeQTLS.sh
    Untracked:  code/._runMakeEmpiricaleQTL_unexp.sh
    Untracked:  code/._runMakeeQTLempirical.sh
    Untracked:  code/._run_getApaPval4eqtl.sh
    Untracked:  code/._run_getapafromeQTL.py
    Untracked:  code/._run_getapafromeQTL.sh
    Untracked:  code/._run_getapapval4eqtl_unexp.sh
    Untracked:  code/._run_leafcutterDiffIso.sh
    Untracked:  code/._run_sepUsagephen.sh
    Untracked:  code/._run_sepgenobychrom.sh
    Untracked:  code/._selectNominalPvalues.py
    Untracked:  code/._sepUsagePhen.py
    Untracked:  code/._sepgenobychrom.py
    Untracked:  code/._snakemakePAS.batch
    Untracked:  code/._snakemakefiltPAS.batch
    Untracked:  code/._sortindexRNAbam.sh
    Untracked:  code/._submit-snakemakePAS.sh
    Untracked:  code/._submit-snakemakefiltPAS.sh
    Untracked:  code/._subsetAPAnotEorPgene.py
    Untracked:  code/._subsetApanoteGene.py
    Untracked:  code/._subsetUnexplainedeQTLs.py
    Untracked:  code/._subset_diffisopheno.py
    Untracked:  code/._subsetpermAPAwithGenelist.py
    Untracked:  code/._subtrachfiveprimeUTR.sh
    Untracked:  code/._subtractExons.sh
    Untracked:  code/._subtractfiveprimeUTR.sh
    Untracked:  code/._tabixSNPS.sh
    Untracked:  code/._utrdms2saf.py
    Untracked:  code/.snakemake/
    Untracked:  code/APAqtl_nominal.err
    Untracked:  code/APAqtl_nominal.out
    Untracked:  code/APAqtl_nominal_39.err
    Untracked:  code/APAqtl_nominal_39.out
    Untracked:  code/APAqtl_nominal_nonNorm.err
    Untracked:  code/APAqtl_nominal_nonNorm.out
    Untracked:  code/APAqtl_permuted.err
    Untracked:  code/APAqtl_permuted.out
    Untracked:  code/ApaQTL_nominalNonnorm.sh
    Untracked:  code/BothFracDTPlot1stintron.err
    Untracked:  code/BothFracDTPlot1stintron.out
    Untracked:  code/BothFracDTPlot4stintron.err
    Untracked:  code/BothFracDTPlot4stintron.out
    Untracked:  code/BothFracDTPlotGeneRegions.err
    Untracked:  code/BothFracDTPlotGeneRegions.out
    Untracked:  code/BothFracDTPlotGeneRegions_norm.err
    Untracked:  code/BothFracDTPlotGeneRegions_norm.out
    Untracked:  code/BothFracDTPlotGeneRegions_normalized.sh
    Untracked:  code/DistPAS2Sig.py
    Untracked:  code/EandPqtl.err
    Untracked:  code/EandPqtl.out
    Untracked:  code/EandPqtls.sh
    Untracked:  code/EncodeRNADTPlotGeneRegions.err
    Untracked:  code/EncodeRNADTPlotGeneRegions.out
    Untracked:  code/FC_NucintornUpandDown.sh
    Untracked:  code/FC_NucintronPASupandDown.err
    Untracked:  code/FC_NucintronPASupandDown.out
    Untracked:  code/FC_UTR.err
    Untracked:  code/FC_UTR.out
    Untracked:  code/FC_UTR.sh
    Untracked:  code/FC_intornUpandDownsteamPAS.sh
    Untracked:  code/FC_intronPASupandDown.err
    Untracked:  code/FC_intronPASupandDown.out
    Untracked:  code/FC_nascent.err
    Untracked:  code/FC_nascentout
    Untracked:  code/FC_nascentseq.sh
    Untracked:  code/FC_newPAS_olddata.err
    Untracked:  code/FC_newPAS_olddata.out
    Untracked:  code/FC_newPeaks_olddata.sh
    Untracked:  code/HMMpermuteTotal.py
    Untracked:  code/HmmPermute.p
    Untracked:  code/HmmPermute.py
    Untracked:  code/LC_samplegroups.py
    Untracked:  code/NascentDTPlotGeneRegions.err
    Untracked:  code/NascentDTPlotGeneRegions.out
    Untracked:  code/NascentDTPlotPAS.err
    Untracked:  code/NascentDTPlotPAS.out
    Untracked:  code/NascentDTPlotPAS_3utr.err
    Untracked:  code/NascentDTPlotPAS_3utr.out
    Untracked:  code/NascentDTPlotPAS_firstintron.err
    Untracked:  code/NascentDTPlotPAS_firstintron.out
    Untracked:  code/NascentDTPlotPAS_intron.err
    Untracked:  code/NascentDTPlotPAS_intron.out
    Untracked:  code/NascentDTPlotPAS_nuc.err
    Untracked:  code/NascentDTPlotPAS_nuc.out
    Untracked:  code/NascentDTPlotPAS_tot.err
    Untracked:  code/NascentDTPlotPAS_tot.out
    Untracked:  code/NascentRNAdtPlot.sh
    Untracked:  code/NascentRNAdtPlot3UTRPAS.sh
    Untracked:  code/NascentRNAdtPlotExcludeFirstintronicPAS.sh
    Untracked:  code/NascentRNAdtPlotFirstintronicPAS.sh
    Untracked:  code/NascentRNAdtPlotNucPAS.sh
    Untracked:  code/NascentRNAdtPlotTotPAS.sh
    Untracked:  code/NascentRNAdtPlotintronicPAS.sh
    Untracked:  code/NascnetRNAdtPlotPAS.sh
    Untracked:  code/NetSeq_fourthintronDT.sh
    Untracked:  code/Nuclear_example.err
    Untracked:  code/Nuclear_example.out
    Untracked:  code/QTL2bed.py
    Untracked:  code/QTL2bed_withstrand.py
    Untracked:  code/README.md
    Untracked:  code/RNABam2BW.err
    Untracked:  code/RNABam2BW.out
    Untracked:  code/RNAbam2bw.sh
    Untracked:  code/Rplots.pdf
    Untracked:  code/Script4NuclearQTLexamples.sh
    Untracked:  code/Script4TotalQTLexamples.sh
    Untracked:  code/TESplots100bp.err
    Untracked:  code/TESplots100bp.out
    Untracked:  code/TESplots100bp.sh
    Untracked:  code/TESplots150bp.err
    Untracked:  code/TESplots150bp.out
    Untracked:  code/TESplots150bp.sh
    Untracked:  code/TESplots200bp.err
    Untracked:  code/TESplots200bp.out
    Untracked:  code/TESplots200bp.sh
    Untracked:  code/Total_example.err
    Untracked:  code/Total_example.out
    Untracked:  code/Untitled
    Untracked:  code/Upstream100Bases_general.py
    Untracked:  code/ZipandTabPheno.sh
    Untracked:  code/aAPAqtl_nominal39ind.sh
    Untracked:  code/apaQTLCorrectPvalMakeQQ_4pc.R
    Untracked:  code/apaQTL_Nominal_4pc.sh
    Untracked:  code/apaQTL_permuted.4pc.sh
    Untracked:  code/apafacetboxplots.R
    Untracked:  code/apaqtlfacetboxplots.R
    Untracked:  code/assignNucIntonpeak2intronlocs.sh
    Untracked:  code/assignPeak2Intronicregion.err
    Untracked:  code/assignPeak2Intronicregion.out
    Untracked:  code/assignTotIntronpeak2intronlocs.sh
    Untracked:  code/assigntotPeak2Intronicregion.err
    Untracked:  code/assigntotPeak2Intronicregion.out
    Untracked:  code/bam2BW_5primemost.sh
    Untracked:  code/bam2bw.err
    Untracked:  code/bam2bw.out
    Untracked:  code/bam2bw_5primemost.err
    Untracked:  code/bam2bw_5primemost.out
    Untracked:  code/bothFracDTplot1stintron.sh
    Untracked:  code/bothFracDTplot4thintron.sh
    Untracked:  code/bothFrac_FC.err
    Untracked:  code/bothFrac_FC.out
    Untracked:  code/bothFrac_FC.sh
    Untracked:  code/changenomQTLres2geneName.py
    Untracked:  code/codingdms2bed.py
    Untracked:  code/convertNominal2SNPLOC.py
    Untracked:  code/correctNomeqtl.R
    Untracked:  code/dag.pdf
    Untracked:  code/dagPAS.pdf
    Untracked:  code/dagfiltPAS.pdf
    Untracked:  code/eQTLgenestestedapa.py
    Untracked:  code/encodeRNADTplots.sh
    Untracked:  code/extractGenotypes.py
    Untracked:  code/extractseqfromqtlfastq.py
    Untracked:  code/fc2leafphen.py
    Untracked:  code/finalPASbed2SAF.py
    Untracked:  code/findbuginpeaks.R
    Untracked:  code/fix4su304corr.py
    Untracked:  code/fix4su604corr.py
    Untracked:  code/fix4sukalisto.py
    Untracked:  code/fixExandUnexeQTL
    Untracked:  code/fixExandUnexeQTL.py
    Untracked:  code/fixFChead_bothfrac.py
    Untracked:  code/fixFChead_summary.py
    Untracked:  code/fixH3k12ac.py
    Untracked:  code/fixRNAhead4corr.py
    Untracked:  code/fixRNAkalisto.py
    Untracked:  code/fixgroupedtranscript.py
    Untracked:  code/fixhead_netseqfc.py
    Untracked:  code/genotypesYRI.gen.proc.keep.vcf.log
    Untracked:  code/genotypesYRI.gen.proc.keep.vcf.recode.vcf
    Untracked:  code/get100upPAS.py
    Untracked:  code/getAPAfromanyeQTL.py
    Untracked:  code/getApapval4eqtl.py
    Untracked:  code/getApapval4eqtl_unexp.py
    Untracked:  code/getDownstreamIntronNuclear.py
    Untracked:  code/getIntronDownstreamPAS.py
    Untracked:  code/getIntronUpstreamPAS.py
    Untracked:  code/getQTLalleles.py
    Untracked:  code/getQTLfastq.sh
    Untracked:  code/getSeq100up.sh
    Untracked:  code/getUpstreamIntronNuclear.py
    Untracked:  code/getseq100up.err
    Untracked:  code/getseq100up.out
    Untracked:  code/grouptranscripts.err
    Untracked:  code/grouptranscripts.out
    Untracked:  code/grouptranscripts.py
    Untracked:  code/keep5perMAF.py
    Untracked:  code/keepSNP_vcf.sh
    Untracked:  code/log/
    Untracked:  code/makeSAFbothfrac5perc.py
    Untracked:  code/makeSNP2rsidfile.py
    Untracked:  code/makeeQTLempirical_unexp.py
    Untracked:  code/makeeQTLempiricaldist.py
    Untracked:  code/makegencondeTSSfile.py
    Untracked:  code/mergRNABam.sh
    Untracked:  code/mergeBW_norm.sh
    Untracked:  code/mergeBWnorm.err
    Untracked:  code/mergeBWnorm.out
    Untracked:  code/mergeBamNacent.err
    Untracked:  code/mergeBamNacent.out
    Untracked:  code/mergeBamNascent.sh
    Untracked:  code/mergeRNAbam.err
    Untracked:  code/mergeRNAbam.out
    Untracked:  code/mnase1stintron.sh
    Untracked:  code/mnaseDTPlot1stintron.err
    Untracked:  code/mnaseDTPlot1stintron.out
    Untracked:  code/mnaseDTPlot4thintron.err
    Untracked:  code/mnaseDTPlot4thintron.out
    Untracked:  code/mnaseDT_fourthintron.sh
    Untracked:  code/netDTPlot4thintron.out
    Untracked:  code/netseqDTplot1stIntron.sh
    Untracked:  code/netseqFC.err
    Untracked:  code/netseqFC.out
    Untracked:  code/netseqFC.sh
    Untracked:  code/neyDTPlot4thintron.err
    Untracked:  code/phenoQTLfromlist.py
    Untracked:  code/processYRIgen.py
    Untracked:  code/qtlFacetBoxplots.err
    Untracked:  code/qtlFacetBoxplots.out
    Untracked:  code/qtlRegionseq.sh
    Untracked:  code/qtlsPvalOppFrac.py
    Untracked:  code/removeXfromHmm.py
    Untracked:  code/removeloc_pheno.py
    Untracked:  code/runCorrectNomEqtl.sh
    Untracked:  code/runCorrectNomeqtl.err
    Untracked:  code/runCorrectNomeqtl.out
    Untracked:  code/runHMMpermute.err
    Untracked:  code/runHMMpermute.out
    Untracked:  code/runHMMpermuteAPAqtls.sh
    Untracked:  code/runHMMpermuteeQTLS.sh
    Untracked:  code/runHMMpermuteeQTLs.err
    Untracked:  code/runHMMpermuteeQTLs.out
    Untracked:  code/runMakeEmpiricaleQTL_unexp.sh
    Untracked:  code/runMakeEmpiricaleQTLs.err
    Untracked:  code/runMakeEmpiricaleQTLs.out
    Untracked:  code/runMakeEmpiricaleQTLsunex.err
    Untracked:  code/runMakeEmpiricaleQTLsunex.out
    Untracked:  code/runMakeeQTLempirical.sh
    Untracked:  code/run_DistPAS2Sig.err
    Untracked:  code/run_DistPAS2Sig.out
    Untracked:  code/run_distPAS2Sig.sh
    Untracked:  code/run_getAPAfromanyeQTL.err
    Untracked:  code/run_getAPAfromanyeQTL.out
    Untracked:  code/run_getApaPval4eQTLs.err
    Untracked:  code/run_getApaPval4eQTLs.out
    Untracked:  code/run_getApaPval4eQTLsunexplained.err
    Untracked:  code/run_getApaPval4eQTLsunexplained.out
    Untracked:  code/run_getApaPval4eqtl.sh
    Untracked:  code/run_getapafromeQTL.sh
    Untracked:  code/run_getapapval4eqtl_unexp.sh
    Untracked:  code/run_leafcutterDiffIso.sh
    Untracked:  code/run_leafcutter_ds.err
    Untracked:  code/run_leafcutter_ds.out
    Untracked:  code/run_qtlFacetBoxplots.sh
    Untracked:  code/run_sepUsagephen.sh
    Untracked:  code/run_sepgenobychrom.err
    Untracked:  code/run_sepgenobychrom.out
    Untracked:  code/run_sepgenobychrom.sh
    Untracked:  code/run_sepusage.err
    Untracked:  code/run_sepusage.out
    Untracked:  code/selectNominalPvalues.py
    Untracked:  code/sepUsagePhen.py
    Untracked:  code/sepgenobychrom.py
    Untracked:  code/seqQTLfastq.err
    Untracked:  code/seqQTLfastq.out
    Untracked:  code/seqQTLregion.err
    Untracked:  code/seqQTLregion.out
    Untracked:  code/snakePASlog.out
    Untracked:  code/snakefiltPASlog.out
    Untracked:  code/sortindexRNABam.err
    Untracked:  code/sortindexRNABam.out
    Untracked:  code/sortindexRNAbam.sh
    Untracked:  code/subsetAPAnotEorPgene.py
    Untracked:  code/subsetApanoteGene.py
    Untracked:  code/subsetUnexplainedeQTLs.py
    Untracked:  code/subset_diffisopheno.py
    Untracked:  code/subsetpermAPAwithGenelist.py
    Untracked:  code/subtract5UTR.err
    Untracked:  code/subtract5UTR.out
    Untracked:  code/subtractExons.err
    Untracked:  code/subtractExons.out
    Untracked:  code/subtractExons.sh
    Untracked:  code/subtractfiveprimeUTR.sh
    Untracked:  code/tabixSNPS.sh
    Untracked:  code/tabixSNPs.err
    Untracked:  code/tabixSNPs.out
    Untracked:  code/transcriptdm2bed.py
    Untracked:  code/utrdms2saf.py
    Untracked:  code/vcf_keepsnps.err
    Untracked:  code/vcf_keepsnps.out
    Untracked:  code/writeExampleQTLcode.py
    Untracked:  code/zipandtabPhen.err
    Untracked:  code/zipandtabPhen.out
    Untracked:  data/._.DS_Store
    Untracked:  data/ApaByEgene/
    Untracked:  data/ApaByPgene/
    Untracked:  data/Battle_pQTL/
    Untracked:  data/CompareOldandNew/
    Untracked:  data/DTmatrix/
    Untracked:  data/DiffIso/
    Untracked:  data/EncodeRNA/
    Untracked:  data/ExampleQTLPlots/
    Untracked:  data/GeuvadisRNA/
    Untracked:  data/HMMqtls/
    Untracked:  data/Li_eQTLs/
    Untracked:  data/NascentRNA/
    Untracked:  data/NucSpeceQTLeffect/
    Untracked:  data/PAS/
    Untracked:  data/QTLGenotypes/
    Untracked:  data/QTLoverlap/
    Untracked:  data/QTLoverlap_nonNorm/
    Untracked:  data/README.md
    Untracked:  data/RNAseq/
    Untracked:  data/Reads2UTR/
    Untracked:  data/SignalSiteFiles/
    Untracked:  data/ThirtyNineIndQtl_nominal/
    Untracked:  data/apaQTLNominal/
    Untracked:  data/apaQTLNominal_4pc/
    Untracked:  data/apaQTLPermuted/
    Untracked:  data/apaQTLPermuted_4pc/
    Untracked:  data/apaQTLs/
    Untracked:  data/assignedPeaks/
    Untracked:  data/bam/
    Untracked:  data/bam_clean/
    Untracked:  data/bam_waspfilt/
    Untracked:  data/bed_10up/
    Untracked:  data/bed_clean/
    Untracked:  data/bed_clean_sort/
    Untracked:  data/bed_waspfilter/
    Untracked:  data/bedsort_waspfilter/
    Untracked:  data/bothFrac_FC/
    Untracked:  data/bw_norm/
    Untracked:  data/eQTLs/
    Untracked:  data/exampleQTLs/
    Untracked:  data/fastq/
    Untracked:  data/filterPeaks/
    Untracked:  data/fourSU/
    Untracked:  data/h3k27ac/
    Untracked:  data/highdiffsiggenes.txt
    Untracked:  data/inclusivePeaks/
    Untracked:  data/inclusivePeaks_FC/
    Untracked:  data/intronRNAratio/
    Untracked:  data/intron_analysis/
    Untracked:  data/mergedBG/
    Untracked:  data/mergedBW_byfrac/
    Untracked:  data/mergedBW_norm/
    Untracked:  data/mergedBam/
    Untracked:  data/mergedbyFracBam/
    Untracked:  data/molPhenos/
    Untracked:  data/molQTLs/
    Untracked:  data/motifdistrupt/
    Untracked:  data/netseq/
    Untracked:  data/nonNorm_pheno/
    Untracked:  data/nuc_10up/
    Untracked:  data/nuc_10upclean/
    Untracked:  data/overlapeQTL_try2/
    Untracked:  data/overlapeQTLs/
    Untracked:  data/peakCoverage/
    Untracked:  data/peaks_5perc/
    Untracked:  data/phenotype/
    Untracked:  data/phenotype_5perc/
    Untracked:  data/sigDiffGenes.txt
    Untracked:  data/sort/
    Untracked:  data/sort_clean/
    Untracked:  data/sort_waspfilter/
    Untracked:  nohup.out
    Untracked:  output/._.DS_Store
    Untracked:  output/._meanCorrelationPhenotypes.svg
    Untracked:  output/dtPlots/
    Untracked:  output/fastqc/
    Untracked:  output/meanCorrelationPhenotypes.svg

Unstaged changes:
    Modified:   analysis/NuclearSpecAPAqtl.Rmd
    Modified:   analysis/Readdistagainstfeatures.Rmd
    Modified:   analysis/overlapapaqtlsandeqtls.Rmd
    Modified:   analysis/propeQTLs_explained.Rmd
    Modified:   code/BothFracDTPlotGeneRegions.sh
    Modified:   code/Snakefile
    Deleted:    code/Upstream10Bases_general.py
    Modified:   code/apaQTLCorrectPvalMakeQQ.R
    Modified:   code/apaQTL_Nominal.sh
    Modified:   code/apaQTL_permuted.sh
    Modified:   code/apaQTLsnake.err
    Modified:   code/bam2bw.sh
    Modified:   code/bed2saf.py
    Modified:   code/cluster.json
    Modified:   code/clusterfiltPAS.json
    Modified:   code/config.yaml
    Modified:   code/environment.yaml
    Modified:   code/makePheno.py
    Deleted:    code/test.txt

</code></pre>
<p>
Note that any generated files, e.g.Â HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If youâ€™ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/438b5c0afe6010686cad87f7c91599966dd48cc6/analysis/nascenttranscription.Rmd" target="_blank">438b5c0</a>
</td>
<td>
brimittleman
</td>
<td>
2019-06-24
</td>
<td>
add npas
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/apaQTL/5b239b1cc630395b3c01ec2160af1b68ee1541e7/docs/nascenttranscription.html" target="_blank">5b239b1</a>
</td>
<td>
brimittleman
</td>
<td>
2019-06-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/5ea9c06fc90df7aef4583b943d1f682e933af3fe/analysis/nascenttranscription.Rmd" target="_blank">5ea9c06</a>
</td>
<td>
brimittleman
</td>
<td>
2019-06-13
</td>
<td>
fix bug
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/apaQTL/7aeba546404bbaa7fa085dbd6062a37a0fb3c143/docs/nascenttranscription.html" target="_blank">7aeba54</a>
</td>
<td>
brimittleman
</td>
<td>
2019-05-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/78b53a1cd0e8e370a6f7087ded2c3a495d3a1f41/analysis/nascenttranscription.Rmd" target="_blank">78b53a1</a>
</td>
<td>
brimittleman
</td>
<td>
2019-05-17
</td>
<td>
add full apa by loc
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/apaQTL/a295d27d552fc98286e043edd5c5f0fbb21f17d2/docs/nascenttranscription.html" target="_blank">a295d27</a>
</td>
<td>
brimittleman
</td>
<td>
2019-05-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/75f456708c9fb69499addbde5ed57b5f6370be83/analysis/nascenttranscription.Rmd" target="_blank">75f4567</a>
</td>
<td>
brimittleman
</td>
<td>
2019-05-16
</td>
<td>
add total intron/all
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/apaQTL/460e1fb1112f135cc0704c5a91d932f16a930450/docs/nascenttranscription.html" target="_blank">460e1fb</a>
</td>
<td>
brimittleman
</td>
<td>
2019-05-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/1df3fe1053fa61acd4d2ca8f6a36287fd5ebf8f6/analysis/nascenttranscription.Rmd" target="_blank">1df3fe1</a>
</td>
<td>
brimittleman
</td>
<td>
2019-05-16
</td>
<td>
seperate fractions by locations
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/apaQTL/81a3e1646c908c34822af2ce80eac2e07ee602d6/docs/nascenttranscription.html" target="_blank">81a3e16</a>
</td>
<td>
brimittleman
</td>
<td>
2019-05-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/f484dcdb4493c72bb05e76cd64d14389de036457/analysis/nascenttranscription.Rmd" target="_blank">f484dcd</a>
</td>
<td>
brimittleman
</td>
<td>
2019-05-15
</td>
<td>
add nascent transcription plot
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(reshape2)
library(workflowr)</code></pre>
<pre><code>This is workflowr version 1.3.0
Run ?workflowr for help getting started</code></pre>
<pre class="r"><code>library(tidyverse)</code></pre>
<pre><code>â”€â”€ Attaching packages â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€ tidyverse 1.2.1 â”€â”€</code></pre>
<pre><code>âœ” ggplot2 3.1.1       âœ” purrr   0.3.2  
âœ” tibble  2.1.1       âœ” dplyr   0.8.0.1
âœ” tidyr   0.8.3       âœ” stringr 1.3.1  
âœ” readr   1.3.1       âœ” forcats 0.3.0  </code></pre>
<pre><code>â”€â”€ Conflicts â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€ tidyverse_conflicts() â”€â”€
âœ– dplyr::filter() masks stats::filter()
âœ– dplyr::lag()    masks stats::lag()</code></pre>
<pre class="r"><code>library(viridis)</code></pre>
<pre><code>Loading required package: viridisLite</code></pre>
<p>Gene name switch file:</p>
<pre class="r"><code>geneNames=read.table(&quot;../../genome_anotation_data/ensemble_to_genename.txt&quot;, sep=&quot;\t&quot;, col.names = c(&#39;gene_id&#39;, &#39;GeneName&#39;, &#39;source&#39; ),stringsAsFactors = F)</code></pre>
<div id="create-transcription-phenotype" class="section level2">
<h2>Create transcription phenotype</h2>
<p>4su data</p>
<pre class="r"><code>FourSU=read.table(file = &quot;../data/fourSU/tr_decay_table_norm.txt&quot;, header=T, stringsAsFactors = F) %&gt;%  dplyr::select(gene_id,contains(&quot;4su_30&quot;))


FourSU_geneNames=FourSU %&gt;% inner_join(geneNames, by=&quot;gene_id&quot;) %&gt;% dplyr::select(GeneName, contains(&quot;4su_30&quot;))

FourgeneNames_long=melt(FourSU_geneNames,id.vars = &quot;GeneName&quot;, value.name = &quot;FourSU&quot;, variable.name = &quot;FourSU_ind&quot;) %&gt;% separate(FourSU_ind, into=c(&quot;type&quot;,&quot;time&quot;, &quot;1400&quot;, &quot;MAf&quot;, &quot;Individual&quot;), sep=&quot;_&quot;) %&gt;% dplyr::select(GeneName, Individual, FourSU) 

FourSU_geneMean=FourgeneNames_long %&gt;% group_by(GeneName) %&gt;%summarise(Mean_4su=mean(FourSU))</code></pre>
<p>rna seq</p>
<pre class="r"><code>RNA=read.table(file = &quot;../data/fourSU/tr_decay_table_norm.txt&quot;, header=T, stringsAsFactors = F) %&gt;%  dplyr::select(gene_id,contains(&quot;RNAseq_14000&quot;))


RNA_geneNames=RNA %&gt;% inner_join(geneNames, by=&quot;gene_id&quot;) %&gt;% dplyr::select(GeneName, contains(&quot;RNA&quot;))

RNAgeneNames_long=melt(RNA_geneNames,id.vars = &quot;GeneName&quot;, value.name = &quot;RNA&quot;, variable.name = &quot;RNA_ind&quot;) %&gt;%   separate(RNA_ind, into=c(&quot;type&quot;, &quot;1400&quot;, &quot;MAf&quot;, &quot;Individual&quot;), sep=&quot;_&quot;) %&gt;% dplyr::select(GeneName, Individual, RNA) 

RNA_geneMean=RNAgeneNames_long %&gt;% group_by(GeneName) %&gt;%summarise(Mean_RNA=mean(RNA))</code></pre>
<p>Make transcription phenotype</p>
<pre class="r"><code>Transcription=FourSU_geneMean %&gt;% inner_join(RNA_geneMean, by=&quot;GeneName&quot;) %&gt;% mutate(Transcription=Mean_4su/(Mean_4su + Mean_RNA)) %&gt;% dplyr::select(GeneName, Transcription) %&gt;% dplyr::rename(&quot;gene&quot;=GeneName)

Transcription2=FourSU_geneMean %&gt;% inner_join(RNA_geneMean, by=&quot;GeneName&quot;) %&gt;% mutate(Transcription=Mean_4su/Mean_RNA) %&gt;% dplyr::select(GeneName, Transcription) %&gt;% dplyr::rename(&quot;gene&quot;=GeneName)</code></pre>
</div>
<div id="apa-phenotype" class="section level2">
<h2>APA phenotype</h2>
<p>5 perc apa</p>
<pre class="r"><code>peaknumlist=read.table(&quot;../data/peaks_5perc/APApeak_Peaks_GeneLocAnno.5perc.bed&quot;, stringsAsFactors = F, header=F, col.names = c(&quot;chr&quot;, &quot;start&quot;,&quot;end&quot;, &quot;id&quot;, &quot;score&quot;, &quot;strand&quot;))  %&gt;% separate(id, into=c(&quot;peaknum&quot;, &quot;geneid&quot;), sep=&quot;:&quot;) %&gt;% mutate(peakid=paste(&quot;peak&quot;, peaknum,sep=&quot;&quot;))</code></pre>
<p>Restrict to genes with large diff between file:</p>
<pre class="r"><code>sig_genes=read.table(file=&quot;../data/highdiffsiggenes.txt&quot;,col.names = &quot;gene&quot;,stringsAsFactors = F)</code></pre>
<p>Nuclear apa</p>
<pre class="r"><code>NucAPA=read.table(&quot;../data/peakCoverage/APAPeaks.ALLChrom.Filtered.Named.GeneLocAnnoPARSED.Nuclear.Quant.Fixed.fc&quot;, stringsAsFactors = F, header = T) %&gt;% dplyr::select(-Chr, -Start, -End, -Strand, -Length) %&gt;% separate(Geneid, into=c(&quot;peakid&quot;,&quot;chrom&quot;, &quot;start&quot;, &quot;end&quot;, &quot;strand&quot;, &quot;geneID&quot;),sep=&quot;:&quot;) %&gt;% semi_join(peaknumlist, by=&quot;peakid&quot;) %&gt;% separate(geneID, into=c(&quot;gene&quot;, &quot;loc&quot;), sep=&quot;_&quot;) %&gt;% dplyr::select(-chrom , -start, -end, -strand, -loc) %&gt;% semi_join(sig_genes, by=&quot;gene&quot;)

NucApaMelt=melt(NucAPA, id.vars =c( &quot;peakid&quot;, &quot;gene&quot;), value.name=&quot;count&quot;, variable.name=&quot;Ind&quot;) %&gt;% separate(Ind, into=c(&#39;Individual&#39;, &#39;fraction&#39;) ,sep=&quot;_&quot;)%&gt;% dplyr::select(peakid, gene, Individual, count)


NucAPA_bygene= NucApaMelt %&gt;% group_by(gene,Individual) %&gt;% summarise(NuclearSum=sum(count))</code></pre>
<p>total apa</p>
<pre class="r"><code>TotAPA=read.table(&quot;../data/peakCoverage/APAPeaks.ALLChrom.Filtered.Named.GeneLocAnnoPARSED.Total.Quant.Fixed.fc&quot;, stringsAsFactors = F, header = T) %&gt;% dplyr::select(-Chr, -Start, -End, -Strand, -Length) %&gt;% separate(Geneid, into=c(&quot;peakid&quot;,&quot;chrom&quot;, &quot;start&quot;, &quot;end&quot;, &quot;strand&quot;, &quot;geneID&quot;),sep=&quot;:&quot;) %&gt;% semi_join(peaknumlist, by=&quot;peakid&quot;) %&gt;% separate(geneID, into=c(&quot;gene&quot;, &quot;loc&quot;), sep=&quot;_&quot;) %&gt;% dplyr::select(-chrom , -start, -end, -strand, -loc) %&gt;% semi_join(sig_genes, by=&quot;gene&quot;)

TotApaMelt=melt(TotAPA, id.vars =c( &quot;peakid&quot;, &quot;gene&quot;), value.name=&quot;count&quot;, variable.name=&quot;Ind&quot;) %&gt;% separate(Ind, into=c(&#39;Individual&#39;, &#39;fraction&#39;) ,sep=&quot;_&quot;)%&gt;% dplyr::select(peakid, gene, Individual, count)


TotAPA_bygene= TotApaMelt %&gt;% group_by(gene,Individual) %&gt;% summarise(TotalSum=sum(count))</code></pre>
<p>Sum together:</p>
<pre class="r"><code>ApaBothFrac=TotAPA_bygene %&gt;% inner_join(NucAPA_bygene, by=c(&quot;gene&quot;, &quot;Individual&quot;))

ApaBothFrac_melt=melt(ApaBothFrac, id.vars=c(&quot;gene&quot;, &quot;Individual&quot;),value.name=&quot;APA_val&quot; ) %&gt;% mutate(fraction=ifelse(variable==&quot;TotalSum&quot;, &quot;total&quot;, &quot;nuclear&quot;), line=paste(&quot;NA&quot;, substring(Individual, 2), sep=&quot;&quot;)) %&gt;% dplyr::select(gene, fraction, line, APA_val)</code></pre>
<p>Normalize with meta data info:</p>
<pre class="r"><code>metadata=read.table(&quot;../data/MetaDataSequencing.txt&quot;, header = T,stringsAsFactors = F) %&gt;% dplyr::select(line, fraction, Mapped_noMP)

metadata$line= as.character(metadata$line)

ApaBothFracStand=ApaBothFrac_melt %&gt;% full_join(metadata, by=c(&quot;line&quot;, &quot;fraction&quot;)) %&gt;% mutate(StandApa=APA_val/Mapped_noMP)

ApaBothFracStand_geneMean=ApaBothFracStand %&gt;% group_by(fraction, gene) %&gt;% summarise(meanAPA=mean(StandApa, na.rm=T))

ApaBothFracStand_geneMean_spread= spread(ApaBothFracStand_geneMean,fraction,meanAPA ) %&gt;% mutate(APAVal=nuclear/(total+ nuclear)) </code></pre>
</div>
<div id="join-data-and-plot" class="section level2">
<h2>Join data and plot</h2>
<p>Density function:</p>
<pre class="r"><code>get_density &lt;- function(x, y, ...) {
  dens &lt;- MASS::kde2d(x, y, ...)
  ix &lt;- findInterval(x, dens$x)
  iy &lt;- findInterval(y, dens$y)
  ii &lt;- cbind(ix, iy)
  return(dens$z[ii])
}

set.seed(1)
dat &lt;- data.frame(
  x = c(
    rnorm(1e4, mean = 0, sd = 0.1),
    rnorm(1e3, mean = 0, sd = 0.1)
  ),
  y = c(
    rnorm(1e4, mean = 0, sd = 0.1),
    rnorm(1e3, mean = 0.1, sd = 0.2)
  )
)</code></pre>
<p>Joing apa and transcription</p>
<pre class="r"><code>APAandTranscrption= Transcription %&gt;% inner_join(ApaBothFracStand_geneMean_spread, by=&quot;gene&quot;)
APAandTranscrption$density &lt;- get_density(APAandTranscrption$APAVal, APAandTranscrption$Transcription, n = 100)



summary(lm(data=APAandTranscrption, APAVal~Transcription))</code></pre>
<pre><code>
Call:
lm(formula = APAVal ~ Transcription, data = APAandTranscrption)

Residuals:
     Min       1Q   Median       3Q      Max 
-0.33795 -0.08278  0.00284  0.08772  0.30261 

Coefficients:
              Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)    0.46672    0.01894  24.641  &lt; 2e-16 ***
Transcription  0.16317    0.03465   4.709 2.88e-06 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 0.1153 on 917 degrees of freedom
Multiple R-squared:  0.02361,   Adjusted R-squared:  0.02254 
F-statistic: 22.17 on 1 and 917 DF,  p-value: 2.877e-06</code></pre>
<p>Plot:</p>
<pre class="r"><code>ggplot(APAandTranscrption, aes(x=Transcription, y=APAVal))+ geom_point(aes(color=density)) + geom_smooth(method = &quot;lm&quot;) + labs(x=&quot;4su/4su+RNA&quot;, y=&quot;Nuclear/Nuclear+Total&quot;, title=&quot;Relationship between APA fraction and transcription&quot;) + scale_color_viridis()</code></pre>
<p><img src="figure/nascenttranscription.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-1">
Past versions of unnamed-chunk-14-1.png
</button>
</p>
<div id="fig-unnamed-chunk-14-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/5b239b1cc630395b3c01ec2160af1b68ee1541e7/docs/figure/nascenttranscription.Rmd/unnamed-chunk-14-1.png" target="_blank">5b239b1</a>
</td>
<td>
brimittleman
</td>
<td>
2019-06-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="split-nuclear-by-intronic" class="section level2">
<h2>Split Nuclear by intronic</h2>
<div id="nuclear-intronic" class="section level3">
<h3>Nuclear intronic:</h3>
<p>I will have to change the gene names for the 3â€™ info:</p>
<pre class="r"><code>NucAPAIntron=read.table(&quot;../data/peakCoverage/APAPeaks.ALLChrom.Filtered.Named.GeneLocAnnoPARSED.Nuclear.Quant.Fixed.fc&quot;, stringsAsFactors = F, header = T) %&gt;% dplyr::select(-Chr, -Start, -End, -Strand, -Length) %&gt;% separate(Geneid, into=c(&quot;peakid&quot;,&quot;chrom&quot;, &quot;start&quot;, &quot;end&quot;, &quot;strand&quot;, &quot;geneID&quot;),sep=&quot;:&quot;) %&gt;% semi_join(peaknumlist, by=&quot;peakid&quot;) %&gt;% separate(geneID, into=c(&quot;gene&quot;, &quot;loc&quot;), sep=&quot;_&quot;) %&gt;% filter(loc==&quot;intron&quot;)%&gt;% dplyr::select(-chrom , -start, -end, -strand, -loc)

NucApaIntronMelt=melt(NucAPAIntron, id.vars =c( &quot;peakid&quot;, &quot;gene&quot;), value.name=&quot;count&quot;, variable.name=&quot;Ind&quot;) %&gt;% separate(Ind, into=c(&#39;Individual&#39;, &#39;fraction&#39;) ,sep=&quot;_&quot;)%&gt;% dplyr::select(peakid, gene, Individual, count)


NucAPAIntron_bygene= NucApaIntronMelt %&gt;% group_by(gene,Individual) %&gt;% summarise(NuclearIntronSum=sum(count))</code></pre>
</div>
<div id="total-utr" class="section level3">
<h3>Total UTR</h3>
<pre class="r"><code>TotUTRAPA=read.table(&quot;../data/peakCoverage/APAPeaks.ALLChrom.Filtered.Named.GeneLocAnnoPARSED.Total.Quant.Fixed.fc&quot;, stringsAsFactors = F, header = T) %&gt;% dplyr::select(-Chr, -Start, -End, -Strand, -Length) %&gt;% separate(Geneid, into=c(&quot;peakid&quot;,&quot;chrom&quot;, &quot;start&quot;, &quot;end&quot;, &quot;strand&quot;, &quot;geneID&quot;),sep=&quot;:&quot;) %&gt;% semi_join(peaknumlist, by=&quot;peakid&quot;) %&gt;% separate(geneID, into=c(&quot;gene&quot;, &quot;loc&quot;), sep=&quot;_&quot;) %&gt;%filter(loc==&quot;utr3&quot;) %&gt;%  dplyr::select(-chrom , -start, -end, -strand, -loc)

TotApaUTRMelt=melt(TotUTRAPA, id.vars =c( &quot;peakid&quot;, &quot;gene&quot;), value.name=&quot;count&quot;, variable.name=&quot;Ind&quot;) %&gt;% separate(Ind, into=c(&#39;Individual&#39;, &#39;fraction&#39;) ,sep=&quot;_&quot;)%&gt;% dplyr::select(peakid, gene, Individual, count)


TotAPAUTR_bygene= TotApaUTRMelt %&gt;% group_by(gene,Individual) %&gt;% summarise(TotalUTRSum=sum(count))</code></pre>
<pre class="r"><code>ApaBothFracLoc=TotAPAUTR_bygene %&gt;% inner_join(NucAPAIntron_bygene, by=c(&quot;gene&quot;, &quot;Individual&quot;))

ApaBothFracLoc_melt=melt(ApaBothFracLoc, id.vars=c(&quot;gene&quot;, &quot;Individual&quot;),value.name=&quot;APA_val&quot; ) %&gt;% mutate(fraction=ifelse(variable==&quot;TotalUTRSum&quot;, &quot;total&quot;, &quot;nuclear&quot;), line=paste(&quot;NA&quot;, substring(Individual, 2), sep=&quot;&quot;)) %&gt;% dplyr::select(gene, fraction, line, APA_val)


ApaBothFracLocStand=ApaBothFracLoc_melt %&gt;% full_join(metadata, by=c(&quot;line&quot;, &quot;fraction&quot;)) %&gt;% mutate(StandApa=APA_val/Mapped_noMP)

ApaBothFracLocStand_geneMean=ApaBothFracLocStand %&gt;% group_by(fraction, gene) %&gt;% summarise(meanAPA=mean(StandApa, na.rm=T))

ApaBothFracLocStand_geneMean_spread= spread(ApaBothFracLocStand_geneMean,fraction,meanAPA ) %&gt;% mutate(APAValLoc=nuclear/(total+nuclear)) 

ApaBothFracLocStand_geneMean_spread2= spread(ApaBothFracLocStand_geneMean,fraction,meanAPA ) %&gt;% mutate(APAValLoc=nuclear/total) </code></pre>
<p>Join this with the transcription info:</p>
<pre class="r"><code>APAlocationandTranscrption= Transcription %&gt;% inner_join(ApaBothFracLocStand_geneMean_spread, by=&quot;gene&quot;)
APAlocationandTranscrption$density &lt;- get_density(APAlocationandTranscrption$APAValLoc, APAlocationandTranscrption$Transcription, n = 100)</code></pre>
<pre class="r"><code>ggplot(APAlocationandTranscrption, aes(x=Transcription, y=APAValLoc))+ geom_point(aes(color=density)) + geom_smooth(method = &quot;lm&quot;) + labs(x=&quot;4su/4su+RNA&quot;, y=&quot;NuclearIntron/TotalUTR + IntronNuclear&quot;, title=&quot;Relationship between APA fraction and transcription&quot;) + scale_color_viridis()</code></pre>
<p><img src="figure/nascenttranscription.Rmd/unnamed-chunk-19-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-19-1">
Past versions of unnamed-chunk-19-1.png
</button>
</p>
<div id="fig-unnamed-chunk-19-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/5b239b1cc630395b3c01ec2160af1b68ee1541e7/docs/figure/nascenttranscription.Rmd/unnamed-chunk-19-1.png" target="_blank">5b239b1</a>
</td>
<td>
brimittleman
</td>
<td>
2019-06-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>summary(lm(data=APAlocationandTranscrption, APAValLoc~Transcription))</code></pre>
<pre><code>
Call:
lm(formula = APAValLoc ~ Transcription, data = APAlocationandTranscrption)

Residuals:
     Min       1Q   Median       3Q      Max 
-0.37748 -0.19928 -0.06962  0.15961  0.73348 

Coefficients:
              Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)    0.13859    0.02161   6.414  1.6e-10 ***
Transcription  0.33661    0.03889   8.656  &lt; 2e-16 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 0.2425 on 3618 degrees of freedom
Multiple R-squared:  0.02029,   Adjusted R-squared:  0.02002 
F-statistic: 74.93 on 1 and 3618 DF,  p-value: &lt; 2.2e-16</code></pre>
<p>Just the ratio:</p>
<pre class="r"><code>APAlocationandTranscrption2= Transcription2 %&gt;% inner_join(ApaBothFracLocStand_geneMean_spread2, by=&quot;gene&quot;)
APAlocationandTranscrption2$density &lt;- get_density(APAlocationandTranscrption2$APAValLoc, APAlocationandTranscrption2$Transcription, n = 100)
summary(lm(data=APAlocationandTranscrption2, log10(APAValLoc)~log10(Transcription)))</code></pre>
<pre><code>
Call:
lm(formula = log10(APAValLoc) ~ log10(Transcription), data = APAlocationandTranscrption2)

Residuals:
     Min       1Q   Median       3Q      Max 
-1.59020 -0.44292 -0.04325  0.39823  2.53804 

Coefficients:
                     Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)          -0.47676    0.01104 -43.202   &lt;2e-16 ***
log10(Transcription)  0.51566    0.05285   9.756   &lt;2e-16 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 0.608 on 3618 degrees of freedom
Multiple R-squared:  0.02563,   Adjusted R-squared:  0.02536 
F-statistic: 95.18 on 1 and 3618 DF,  p-value: &lt; 2.2e-16</code></pre>
<pre class="r"><code>ggplot(APAlocationandTranscrption2, aes(x=log10(Transcription), y=log10(APAValLoc)))+ geom_point(aes(color=density)) + geom_smooth(method = &quot;lm&quot;) + labs(x=&quot;log10(4su/RNA)&quot;, y=&quot;log10(NuclearIntron/TotalUTR)&quot;, title=&quot;Relationship between APA fraction and transcription&quot;) + scale_color_viridis()</code></pre>
<p><img src="figure/nascenttranscription.Rmd/unnamed-chunk-21-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-21-1">
Past versions of unnamed-chunk-21-1.png
</button>
</p>
<div id="fig-unnamed-chunk-21-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/5b239b1cc630395b3c01ec2160af1b68ee1541e7/docs/figure/nascenttranscription.Rmd/unnamed-chunk-21-1.png" target="_blank">5b239b1</a>
</td>
<td>
brimittleman
</td>
<td>
2019-06-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="compare-nuclear-and-total-utr" class="section level2">
<h2>Compare nuclear and total UTR</h2>
<pre class="r"><code>NucAPAUTR=read.table(&quot;../data/peakCoverage/APAPeaks.ALLChrom.Filtered.Named.GeneLocAnnoPARSED.Nuclear.Quant.Fixed.fc&quot;, stringsAsFactors = F, header = T) %&gt;% dplyr::select(-Chr, -Start, -End, -Strand, -Length) %&gt;% separate(Geneid, into=c(&quot;peakid&quot;,&quot;chrom&quot;, &quot;start&quot;, &quot;end&quot;, &quot;strand&quot;, &quot;geneID&quot;),sep=&quot;:&quot;) %&gt;% semi_join(peaknumlist, by=&quot;peakid&quot;) %&gt;% separate(geneID, into=c(&quot;gene&quot;, &quot;loc&quot;), sep=&quot;_&quot;) %&gt;% filter(loc==&quot;utr3&quot;)%&gt;% dplyr::select(-chrom , -start, -end, -strand, -loc)

NucAPAUTRMelt=melt(NucAPAUTR, id.vars =c( &quot;peakid&quot;, &quot;gene&quot;), value.name=&quot;count&quot;, variable.name=&quot;Ind&quot;) %&gt;% separate(Ind, into=c(&#39;Individual&#39;, &#39;fraction&#39;) ,sep=&quot;_&quot;)%&gt;% dplyr::select(peakid, gene, Individual, count)


NucAPAUTR_bygene= NucAPAUTRMelt %&gt;% group_by(gene,Individual) %&gt;% summarise(NuclearUTRSum=sum(count))</code></pre>
<pre class="r"><code>ApaBothFracUTR=TotAPAUTR_bygene %&gt;% inner_join(NucAPAUTR_bygene, by=c(&quot;gene&quot;, &quot;Individual&quot;))

ApaBothFracUTR_melt=melt(ApaBothFracUTR, id.vars=c(&quot;gene&quot;, &quot;Individual&quot;),value.name=&quot;APA_val&quot; ) %&gt;% mutate(fraction=ifelse(variable==&quot;TotalUTRSum&quot;, &quot;total&quot;, &quot;nuclear&quot;), line=paste(&quot;NA&quot;, substring(Individual, 2), sep=&quot;&quot;)) %&gt;% dplyr::select(gene, fraction, line, APA_val)


ApaBothFracUTRStand=ApaBothFracUTR_melt %&gt;% full_join(metadata, by=c(&quot;line&quot;, &quot;fraction&quot;)) %&gt;% mutate(StandApa=APA_val/Mapped_noMP)

ApaBothFracUTRStand_geneMean=ApaBothFracUTRStand %&gt;% group_by(fraction, gene) %&gt;% summarise(meanAPA=mean(StandApa, na.rm=T))

ApaBothFracUTRStand_geneMean_spread= spread(ApaBothFracUTRStand_geneMean,fraction,meanAPA ) %&gt;% mutate(APAValLoc=nuclear/total)
ApaBothFracUTRStand_geneMean_spread2= spread(ApaBothFracUTRStand_geneMean,fraction,meanAPA ) %&gt;% mutate(APAValLoc=nuclear/(total+nuclear))</code></pre>
<p>THis is nuclear vs total only looking at teh UTR:</p>
<pre class="r"><code>APAUTRandTranscrption= Transcription %&gt;% inner_join(ApaBothFracUTRStand_geneMean_spread, by=&quot;gene&quot;)
APAUTRandTranscrption$density &lt;- get_density(APAUTRandTranscrption$APAValLoc, APAUTRandTranscrption$Transcription, n = 100)


summary(lm(data=APAUTRandTranscrption, log10(APAValLoc)~log10(Transcription)))</code></pre>
<pre><code>
Call:
lm(formula = log10(APAValLoc) ~ log10(Transcription), data = APAUTRandTranscrption)

Residuals:
     Min       1Q   Median       3Q      Max 
-0.69129 -0.18916 -0.00344  0.19699  0.90698 

Coefficients:
                     Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)          0.153728   0.008935   17.20   &lt;2e-16 ***
log10(Transcription) 0.481833   0.029049   16.59   &lt;2e-16 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 0.2526 on 7709 degrees of freedom
Multiple R-squared:  0.03446,   Adjusted R-squared:  0.03433 
F-statistic: 275.1 on 1 and 7709 DF,  p-value: &lt; 2.2e-16</code></pre>
<pre class="r"><code>ggplot(APAUTRandTranscrption, aes(x=Transcription, y=APAValLoc))+ geom_point(aes(color=density)) + geom_smooth(method = &quot;lm&quot;) + labs(x=&quot;4su/RNA&quot;, y=&quot;NuclearUTR/TotalUTR&quot;, title=&quot;Relationship between APA fraction and transcription&quot;) + scale_color_viridis()</code></pre>
<p><img src="figure/nascenttranscription.Rmd/unnamed-chunk-24-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-24-1">
Past versions of unnamed-chunk-24-1.png
</button>
</p>
<div id="fig-unnamed-chunk-24-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/5b239b1cc630395b3c01ec2160af1b68ee1541e7/docs/figure/nascenttranscription.Rmd/unnamed-chunk-24-1.png" target="_blank">5b239b1</a>
</td>
<td>
brimittleman
</td>
<td>
2019-06-13
</td>
</tr>
<tr>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/460e1fb1112f135cc0704c5a91d932f16a930450/docs/figure/nascenttranscription.Rmd/unnamed-chunk-24-1.png" target="_blank">460e1fb</a>
</td>
<td>
brimittleman
</td>
<td>
2019-05-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>APAUTRandTranscrption2= Transcription %&gt;% inner_join(ApaBothFracUTRStand_geneMean_spread2, by=&quot;gene&quot;)
APAUTRandTranscrption2$density &lt;- get_density(APAUTRandTranscrption2$APAValLoc, APAUTRandTranscrption2$Transcription, n = 100)


summary(lm(data=APAUTRandTranscrption2, log10(APAValLoc)~log10(Transcription)))</code></pre>
<pre><code>
Call:
lm(formula = log10(APAValLoc) ~ log10(Transcription), data = APAUTRandTranscrption2)

Residuals:
     Min       1Q   Median       3Q      Max 
-0.45003 -0.08569  0.01581  0.10162  0.37243 

Coefficients:
                      Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)          -0.234761   0.004493  -52.25   &lt;2e-16 ***
log10(Transcription)  0.268002   0.014607   18.35   &lt;2e-16 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 0.127 on 7709 degrees of freedom
Multiple R-squared:  0.04184,   Adjusted R-squared:  0.04171 
F-statistic: 336.6 on 1 and 7709 DF,  p-value: &lt; 2.2e-16</code></pre>
<pre class="r"><code>ggplot(APAUTRandTranscrption2, aes(x=Transcription, y=APAValLoc))+ geom_point(aes(color=density)) + geom_smooth(method = &quot;lm&quot;) + labs(x=&quot;4su/RNA+4su&quot;, y=&quot;NuclearUTR/TotalUTR+NuclearUTR&quot;, title=&quot;Relationship between APA fraction and transcription&quot;) + scale_color_viridis()</code></pre>
<p><img src="figure/nascenttranscription.Rmd/unnamed-chunk-25-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-25-1">
Past versions of unnamed-chunk-25-1.png
</button>
</p>
<div id="fig-unnamed-chunk-25-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/5b239b1cc630395b3c01ec2160af1b68ee1541e7/docs/figure/nascenttranscription.Rmd/unnamed-chunk-25-1.png" target="_blank">5b239b1</a>
</td>
<td>
brimittleman
</td>
<td>
2019-06-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="intron-nuclear-over-all-nuclear" class="section level2">
<h2>Intron Nuclear over all nuclear</h2>
<p>Nuclear intron= NucAPAIntron_bygene</p>
<p>all nuclear =NucAPA_bygene</p>
<p>Create this pheno:</p>
<pre class="r"><code>ApaNuclear_byloc=NucAPAIntron_bygene %&gt;% inner_join(NucAPA_bygene, by=c(&quot;gene&quot;, &quot;Individual&quot;)) %&gt;% mutate(IntronOverAll=NuclearIntronSum/NuclearSum) %&gt;% mutate(fraction=&quot;nuclear&quot;,line=paste(&quot;NA&quot;, substring(Individual, 2), sep=&quot;&quot;)) %&gt;% dplyr::select(gene, fraction, line, IntronOverAll) %&gt;% group_by(gene) %&gt;% filter(IntronOverAll!=0) %&gt;%  summarise(MeanIntronoverAll=mean(IntronOverAll)) %&gt;% dplyr::rename(&quot;GeneName&quot;=gene)</code></pre>
<p>Join with RNA</p>
<pre class="r"><code>nuclearandRNA=ApaNuclear_byloc %&gt;% inner_join(RNA_geneMean, by=&quot;GeneName&quot;)

nuclearandRNA$density &lt;- get_density(nuclearandRNA$MeanIntronoverAll, nuclearandRNA$Mean_RNA, n = 100)</code></pre>
<p>Plot:</p>
<pre class="r"><code>ggplot(nuclearandRNA, aes(x=log10(Mean_RNA), y=MeanIntronoverAll))+ geom_point(aes(color=density)) + geom_smooth(method = &quot;lm&quot;) + labs(x=&quot;log10(RNA)&quot;, y=&quot;NuclearIntron/NuclearAll&quot;, title=&quot;Relationship between APA fraction and transcription&quot;) + scale_color_viridis()</code></pre>
<p><img src="figure/nascenttranscription.Rmd/unnamed-chunk-28-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-28-1">
Past versions of unnamed-chunk-28-1.png
</button>
</p>
<div id="fig-unnamed-chunk-28-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/5b239b1cc630395b3c01ec2160af1b68ee1541e7/docs/figure/nascenttranscription.Rmd/unnamed-chunk-28-1.png" target="_blank">5b239b1</a>
</td>
<td>
brimittleman
</td>
<td>
2019-06-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>summary(lm(data=nuclearandRNA, MeanIntronoverAll~log10(Mean_RNA)))</code></pre>
<pre><code>
Call:
lm(formula = MeanIntronoverAll ~ log10(Mean_RNA), data = nuclearandRNA)

Residuals:
     Min       1Q   Median       3Q      Max 
-0.47913 -0.12703 -0.01765  0.12618  0.55620 

Coefficients:
                Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)      0.18933    0.06799   2.784  0.00549 ** 
log10(Mean_RNA) -0.06897    0.01559  -4.425  1.1e-05 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 0.1865 on 811 degrees of freedom
Multiple R-squared:  0.02357,   Adjusted R-squared:  0.02237 
F-statistic: 19.58 on 1 and 811 DF,  p-value: 1.098e-05</code></pre>
<div id="same-plot-with-transcription-phenotype-on-bottom" class="section level3">
<h3>Same plot with transcription phenotype on bottom:</h3>
<pre class="r"><code>ApaNuclear_byloc_rename=ApaNuclear_byloc %&gt;% dplyr::rename(&quot;gene&quot;=GeneName)
nuclearandtranscription=ApaNuclear_byloc_rename %&gt;% inner_join(Transcription, by=&quot;gene&quot;)

nuclearandtranscription$density &lt;- get_density(nuclearandtranscription$MeanIntronoverAll, nuclearandtranscription$Transcription, n = 100)

ggplot(nuclearandtranscription, aes(x=Transcription, y=MeanIntronoverAll))+ geom_point(aes(color=density)) + geom_smooth(method = &quot;lm&quot;) + labs(x=&quot;4su/4su+RNA&quot;, y=&quot;NuclearIntron/NuclearAll&quot;, title=&quot;Relationship between APA fraction and transcription&quot;) + scale_color_viridis()</code></pre>
<p><img src="figure/nascenttranscription.Rmd/unnamed-chunk-30-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-30-1">
Past versions of unnamed-chunk-30-1.png
</button>
</p>
<div id="fig-unnamed-chunk-30-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/5b239b1cc630395b3c01ec2160af1b68ee1541e7/docs/figure/nascenttranscription.Rmd/unnamed-chunk-30-1.png" target="_blank">5b239b1</a>
</td>
<td>
brimittleman
</td>
<td>
2019-06-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>summary(lm(data=nuclearandtranscription, MeanIntronoverAll~Transcription))</code></pre>
<pre><code>
Call:
lm(formula = MeanIntronoverAll ~ Transcription, data = nuclearandtranscription)

Residuals:
     Min       1Q   Median       3Q      Max 
-0.44181 -0.13029 -0.01467  0.12672  0.51883 

Coefficients:
              Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)    0.51462    0.03432  14.994   &lt;2e-16 ***
Transcription -0.04813    0.06269  -0.768    0.443    
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 0.1887 on 811 degrees of freedom
Multiple R-squared:  0.0007261, Adjusted R-squared:  -0.000506 
F-statistic: 0.5893 on 1 and 811 DF,  p-value: 0.4429</code></pre>
</div>
</div>
<div id="intron-total-over-all-total" class="section level2">
<h2>Intron Total over all Total</h2>
<p>First I need to get the total intronic:</p>
<pre class="r"><code>TotAPAIntron=read.table(&quot;../data/peakCoverage/APAPeaks.ALLChrom.Filtered.Named.GeneLocAnnoPARSED.Total.Quant.Fixed.fc&quot;, stringsAsFactors = F, header = T) %&gt;% dplyr::select(-Chr, -Start, -End, -Strand, -Length) %&gt;% separate(Geneid, into=c(&quot;peakid&quot;,&quot;chrom&quot;, &quot;start&quot;, &quot;end&quot;, &quot;strand&quot;, &quot;geneID&quot;),sep=&quot;:&quot;) %&gt;% semi_join(peaknumlist, by=&quot;peakid&quot;) %&gt;% separate(geneID, into=c(&quot;gene&quot;, &quot;loc&quot;), sep=&quot;_&quot;) %&gt;%filter(loc==&quot;intron&quot;) %&gt;%  dplyr::select(-chrom , -start, -end, -strand, -loc)

TotAPAIntronMelt=melt(TotAPAIntron, id.vars =c( &quot;peakid&quot;, &quot;gene&quot;), value.name=&quot;count&quot;, variable.name=&quot;Ind&quot;) %&gt;% separate(Ind, into=c(&#39;Individual&#39;, &#39;fraction&#39;) ,sep=&quot;_&quot;)%&gt;% dplyr::select(peakid, gene, Individual, count)


TotAPAIntron_bygene= TotAPAIntronMelt %&gt;% group_by(gene,Individual) %&gt;% summarise(TotalIntronSum=sum(count))</code></pre>
<pre class="r"><code>ApaTotal_byloc=TotAPAIntron_bygene %&gt;% inner_join(TotAPA_bygene, by=c(&quot;gene&quot;, &quot;Individual&quot;)) %&gt;% mutate(IntronOverAll=TotalIntronSum/TotalSum) %&gt;% mutate(fraction=&quot;total&quot;,line=paste(&quot;NA&quot;, substring(Individual, 2), sep=&quot;&quot;)) %&gt;% dplyr::select(gene, fraction, line, IntronOverAll) %&gt;% group_by(gene) %&gt;% filter(IntronOverAll!=0) %&gt;%  summarise(MeanIntronoverAll=mean(IntronOverAll)) %&gt;% dplyr::rename(&quot;GeneName&quot;=gene)</code></pre>
<p>Join with RNA</p>
<pre class="r"><code>totalandRNA=ApaTotal_byloc %&gt;% inner_join(RNA_geneMean, by=&quot;GeneName&quot;)

totalandRNA$density &lt;- get_density(totalandRNA$MeanIntronoverAll, totalandRNA$Mean_RNA, n = 100)</code></pre>
<p>Plot:</p>
<pre class="r"><code>ggplot(totalandRNA, aes(x=log10(Mean_RNA), y=MeanIntronoverAll))+ geom_point(aes(color=density)) + geom_smooth(method = &quot;lm&quot;) + labs(x=&quot;log10(RNA)&quot;, y=&quot;TotalIntron/TotalAll&quot;, title=&quot;Relationship between APA fraction and transcription&quot;) + scale_color_viridis()</code></pre>
<p><img src="figure/nascenttranscription.Rmd/unnamed-chunk-34-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-34-1">
Past versions of unnamed-chunk-34-1.png
</button>
</p>
<div id="fig-unnamed-chunk-34-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/5b239b1cc630395b3c01ec2160af1b68ee1541e7/docs/figure/nascenttranscription.Rmd/unnamed-chunk-34-1.png" target="_blank">5b239b1</a>
</td>
<td>
brimittleman
</td>
<td>
2019-06-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>summary(lm(data=totalandRNA, MeanIntronoverAll~log10(Mean_RNA)))</code></pre>
<pre><code>
Call:
lm(formula = MeanIntronoverAll ~ log10(Mean_RNA), data = totalandRNA)

Residuals:
     Min       1Q   Median       3Q      Max 
-0.27714 -0.12468 -0.04078  0.06442  0.82065 

Coefficients:
                Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)     -0.32230    0.06483  -4.972  8.1e-07 ***
log10(Mean_RNA) -0.13596    0.01486  -9.149  &lt; 2e-16 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 0.1779 on 811 degrees of freedom
Multiple R-squared:  0.09355,   Adjusted R-squared:  0.09243 
F-statistic:  83.7 on 1 and 811 DF,  p-value: &lt; 2.2e-16</code></pre>
<div id="same-plot-with-transcription-phenotype-on-bottom-1" class="section level3">
<h3>Same plot with transcription phenotype on bottom:</h3>
<pre class="r"><code>ApaTotal_byloc_rename=ApaTotal_byloc %&gt;% dplyr::rename(&quot;gene&quot;=GeneName)
totalandtranscription=ApaTotal_byloc_rename %&gt;% inner_join(Transcription, by=&quot;gene&quot;)

totalandtranscription$density &lt;- get_density(totalandtranscription$MeanIntronoverAll, totalandtranscription$Transcription, n = 100)

ggplot(totalandtranscription, aes(x=Transcription, y=MeanIntronoverAll))+ geom_point(aes(color=density)) + geom_smooth(method = &quot;lm&quot;) + labs(x=&quot;4su/4su+RNA&quot;, y=&quot;TotalIntron/TotalAll&quot;, title=&quot;Relationship between APA fraction and transcription&quot;) + scale_color_viridis()</code></pre>
<p><img src="figure/nascenttranscription.Rmd/unnamed-chunk-36-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-36-1">
Past versions of unnamed-chunk-36-1.png
</button>
</p>
<div id="fig-unnamed-chunk-36-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/5b239b1cc630395b3c01ec2160af1b68ee1541e7/docs/figure/nascenttranscription.Rmd/unnamed-chunk-36-1.png" target="_blank">5b239b1</a>
</td>
<td>
brimittleman
</td>
<td>
2019-06-13
</td>
</tr>
<tr>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/7aeba546404bbaa7fa085dbd6062a37a0fb3c143/docs/figure/nascenttranscription.Rmd/unnamed-chunk-36-1.png" target="_blank">7aeba54</a>
</td>
<td>
brimittleman
</td>
<td>
2019-05-17
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>summary(lm(data=totalandtranscription, MeanIntronoverAll~Transcription))</code></pre>
<pre><code>
Call:
lm(formula = MeanIntronoverAll ~ Transcription, data = totalandtranscription)

Residuals:
     Min       1Q   Median       3Q      Max 
-0.25236 -0.13687 -0.04965  0.08509  0.73980 

Coefficients:
              Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)    0.24548    0.03396   7.227 1.14e-12 ***
Transcription  0.04196    0.06204   0.676    0.499    
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 0.1868 on 811 degrees of freedom
Multiple R-squared:  0.0005638, Adjusted R-squared:  -0.0006685 
F-statistic: 0.4575 on 1 and 811 DF,  p-value: 0.499</code></pre>
</div>
</div>
<div id="put-it-all-together" class="section level2">
<h2>Put it all together:</h2>
<p>(Nuclear intronic/nuclear all)/(total intronic/total all) vs 4su/(4su+RNA)</p>
<p>Nucintron v nuc all:</p>
<p>ApaNuclear_byloc_rename</p>
<p>ApaTotal_byloc_rename</p>
<p>Transcription</p>
<pre class="r"><code>fullapa=ApaNuclear_byloc_rename %&gt;% dplyr::rename(&quot;NuclearIntronoverall&quot;=MeanIntronoverAll)%&gt;% inner_join(ApaTotal_byloc_rename, by=&quot;gene&quot;) %&gt;% mutate(fullAPA=NuclearIntronoverall/MeanIntronoverAll) %&gt;% dplyr::select(gene,fullAPA)


#join with transcription 


BothlocPhenoandtranscription=fullapa %&gt;% inner_join(Transcription, by=&quot;gene&quot;)

BothlocPhenoandtranscription$density &lt;- get_density(BothlocPhenoandtranscription$fullAPA, BothlocPhenoandtranscription$Transcription, n = 100)

ggplot(BothlocPhenoandtranscription, aes(x=Transcription, y=fullAPA))+ geom_point(aes(color=density)) + geom_smooth(method = &quot;lm&quot;) + labs(x=&quot;4su/4su+RNA&quot;, y=&quot;log10(NuclearIntron/Nuclearall)/(TotalIntron/TotalAll)&quot;, title=&quot;Relationship between APA fraction and transcription&quot;) + scale_color_viridis()</code></pre>
<p><img src="figure/nascenttranscription.Rmd/unnamed-chunk-37-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-37-1">
Past versions of unnamed-chunk-37-1.png
</button>
</p>
<div id="fig-unnamed-chunk-37-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/5b239b1cc630395b3c01ec2160af1b68ee1541e7/docs/figure/nascenttranscription.Rmd/unnamed-chunk-37-1.png" target="_blank">5b239b1</a>
</td>
<td>
brimittleman
</td>
<td>
2019-06-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>summary(lm(data=BothlocPhenoandtranscription, log10(fullAPA)~Transcription))</code></pre>
<pre><code>
Call:
lm(formula = log10(fullAPA) ~ Transcription, data = BothlocPhenoandtranscription)

Residuals:
     Min       1Q   Median       3Q      Max 
-0.84376 -0.12135 -0.00255  0.12460  0.74230 

Coefficients:
              Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)    0.48793    0.03819  12.777  &lt; 2e-16 ***
Transcription -0.29941    0.06975  -4.292 1.98e-05 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 0.21 on 811 degrees of freedom
Multiple R-squared:  0.02221,   Adjusted R-squared:  0.02101 
F-statistic: 18.42 on 1 and 811 DF,  p-value: 1.981e-05</code></pre>
</div>
<div id="does-number-of-pas-correlate-with-nascent-transcription" class="section level2">
<h2>Does number of PAS correlate with nascent transcription</h2>
<p>I want to look at number of PAS in the nuclear fraction against 4su/4su+RNA. This is the transcription phenotype in this analysis</p>
<pre class="r"><code>nPASnuc=read.table(&quot;../data/peaks_5perc/APApeak_Peaks_GeneLocAnno.Nuclear.5perc.fc&quot;, header = F, stringsAsFactors = F, col.names = c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;,&quot;gene&quot;, &quot;loc&quot;,&quot;strand&quot;, &quot;peakNum&quot;, &quot;meanusage&quot;)) %&gt;% group_by(gene) %&gt;% summarise(nPASN=n())

transcriptionPAS=Transcription %&gt;%inner_join(nPASnuc, by=&quot;gene&quot;)


transcriptionPAS$density &lt;- get_density(transcriptionPAS$nPASN, transcriptionPAS$Transcription, n = 100)



ggplot(transcriptionPAS, aes(x=Transcription, y=nPASN)) + geom_point(aes(color=density)) + scale_color_viridis() + geom_smooth(method = &quot;lm&quot;) + labs(y=&quot;Number of PAS&quot;, title=&quot;Number of Nuclear PAS increases as nascent transcription increases&quot;) </code></pre>
<p><img src="figure/nascenttranscription.Rmd/unnamed-chunk-38-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>summary(lm(data=transcriptionPAS, nPASN~Transcription))</code></pre>
<pre><code>
Call:
lm(formula = nPASN ~ Transcription, data = transcriptionPAS)

Residuals:
    Min      1Q  Median      3Q     Max 
-3.8788 -1.2951 -0.3395  1.0042  8.4575 

Coefficients:
              Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)    0.38595    0.09275   4.161  3.2e-05 ***
Transcription  4.96412    0.17301  28.692  &lt; 2e-16 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 1.709 on 7885 degrees of freedom
Multiple R-squared:  0.09454,   Adjusted R-squared:  0.09442 
F-statistic: 823.2 on 1 and 7885 DF,  p-value: &lt; 2.2e-16</code></pre>
<p>Try for total fraction:</p>
<pre class="r"><code>nPAStot=read.table(&quot;../data/peaks_5perc/APApeak_Peaks_GeneLocAnno.Total.5perc.fc&quot;, header = F, stringsAsFactors = F, col.names = c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;,&quot;gene&quot;, &quot;loc&quot;,&quot;strand&quot;, &quot;peakNum&quot;, &quot;meanusage&quot;)) %&gt;% group_by(gene) %&gt;% summarise(nPAST=n())

transcriptionTOTPAS=Transcription %&gt;%inner_join(nPAStot, by=&quot;gene&quot;)


transcriptionTOTPAS$density &lt;- get_density(transcriptionTOTPAS$nPAST, transcriptionTOTPAS$Transcription, n = 100)



ggplot(transcriptionTOTPAS, aes(x=Transcription, y=nPAST)) + geom_point(aes(color=density)) + scale_color_viridis() + geom_smooth(method = &quot;lm&quot;) + labs(y=&quot;Number of PAS&quot;, title=&quot;Number of Total PAS increases as nascent transcription increases&quot;) </code></pre>
<p><img src="figure/nascenttranscription.Rmd/unnamed-chunk-39-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>summary(lm(data=transcriptionTOTPAS, nPAST~Transcription))</code></pre>
<pre><code>
Call:
lm(formula = nPAST ~ Transcription, data = transcriptionTOTPAS)

Residuals:
    Min      1Q  Median      3Q     Max 
-3.3359 -1.0608 -0.2954  0.8024  6.7777 

Coefficients:
              Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)    0.15080    0.07847   1.922   0.0547 .  
Transcription  4.62411    0.14637  31.591   &lt;2e-16 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 1.446 on 7885 degrees of freedom
Multiple R-squared:  0.1124,    Adjusted R-squared:  0.1122 
F-statistic:   998 on 1 and 7885 DF,  p-value: &lt; 2.2e-16</code></pre>
<p>Look at the difference in PAS number:</p>
<pre class="r"><code>nPASall= nPAStot %&gt;% inner_join(nPASnuc, by=&quot;gene&quot;) %&gt;% mutate(Difference= nPASN-nPAST) 

transcriptionallPAS=Transcription %&gt;%inner_join(nPASall, by=&quot;gene&quot;)

transcriptionallPAS$density &lt;- get_density(transcriptionallPAS$Difference, transcriptionallPAS$Transcription, n = 100)

ggplot(transcriptionallPAS, aes(x=Transcription, y=Difference)) + geom_point(aes(color=density)) + scale_color_viridis() + geom_smooth(method = &quot;lm&quot;) + labs(y=&quot;Diffference in number of PAS&quot;, title=&quot;Number of Total PAS increases as nascent transcription increases&quot;) </code></pre>
<p><img src="figure/nascenttranscription.Rmd/unnamed-chunk-40-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>summary(lm(data=transcriptionallPAS, Difference~Transcription))</code></pre>
<pre><code>
Call:
lm(formula = Difference ~ Transcription, data = transcriptionallPAS)

Residuals:
    Min      1Q  Median      3Q     Max 
-4.4613 -0.4299 -0.3865  0.5733  6.5790 

Coefficients:
              Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)    0.23515    0.05423   4.336 1.47e-05 ***
Transcription  0.34001    0.10116   3.361  0.00078 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 0.9991 on 7885 degrees of freedom
Multiple R-squared:  0.001431,  Adjusted R-squared:  0.001304 
F-statistic:  11.3 on 1 and 7885 DF,  p-value: 0.0007798</code></pre>
<p>This has a small positive correlation.</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.5.1 (2018-07-02)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Scientific Linux 7.4 (Nitrogen)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] viridis_0.5.1     viridisLite_0.3.0 forcats_0.3.0    
 [4] stringr_1.3.1     dplyr_0.8.0.1     purrr_0.3.2      
 [7] readr_1.3.1       tidyr_0.8.3       tibble_2.1.1     
[10] ggplot2_3.1.1     tidyverse_1.2.1   workflowr_1.3.0  
[13] reshape2_1.4.3   

loaded via a namespace (and not attached):
 [1] tidyselect_0.2.5 haven_1.1.2      lattice_0.20-38  colorspace_1.3-2
 [5] generics_0.0.2   htmltools_0.3.6  yaml_2.2.0       rlang_0.3.1     
 [9] pillar_1.3.1     glue_1.3.0       withr_2.1.2      modelr_0.1.2    
[13] readxl_1.1.0     plyr_1.8.4       munsell_0.5.0    gtable_0.2.0    
[17] cellranger_1.1.0 rvest_0.3.2      evaluate_0.12    labeling_0.3    
[21] knitr_1.20       broom_0.5.1      Rcpp_1.0.0       scales_1.0.0    
[25] backports_1.1.2  jsonlite_1.6     fs_1.2.6         gridExtra_2.3   
[29] hms_0.4.2        digest_0.6.18    stringi_1.2.4    grid_3.5.1      
[33] rprojroot_1.3-2  cli_1.0.1        tools_3.5.1      magrittr_1.5    
[37] lazyeval_0.2.1   crayon_1.3.4     whisker_0.3-2    pkgconfig_2.0.2 
[41] MASS_7.3-51.1    xml2_1.2.0       lubridate_1.7.4  assertthat_0.2.0
[45] rmarkdown_1.10   httr_1.3.1       rstudioapi_0.10  R6_2.3.0        
[49] nlme_3.1-137     git2r_0.25.2     compiler_3.5.1  </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
