<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Briana Mittleman" />


<title>Compare new and old peaks</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/journal.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.0.13/css/fa-svg-with-js.css" rel="stylesheet" />
<script src="site_libs/font-awesome-5.0.13/js/fontawesome-all.min.js"></script>
<script src="site_libs/font-awesome-5.0.13/js/fa-v4-shims.min.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">apaQTL</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https/github.com/brimittleman/apaQTL">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Compare new and old peaks</h1>
<h4 class="author"><em>Briana Mittleman</em></h4>
<h4 class="date"><em>4/30/2019</em></h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2019-05-02
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>apaQTL/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.3.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity itâ€™s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20190411code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20190411)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20190411code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20190411)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g.Â subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcombrimittlemanapaQTLtree0f39c9b26884cd39b2a6647f77e5969736dc3128targetblank0f39c9ba"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/brimittleman/apaQTL/tree/0f39c9b26884cd39b2a6647f77e5969736dc3128" target="_blank">0f39c9b</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcombrimittlemanapaQTLtree0f39c9b26884cd39b2a6647f77e5969736dc3128targetblank0f39c9ba" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    output/.DS_Store

Untracked files:
    Untracked:  .Rprofile
    Untracked:  ._.DS_Store
    Untracked:  .gitignore
    Untracked:  _workflowr.yml
    Untracked:  analysis/._PASdescriptiveplots.Rmd
    Untracked:  analysis/._cuttoffPercUsage.Rmd
    Untracked:  analysis/cuttoffPercUsage.Rmd
    Untracked:  apaQTL.Rproj
    Untracked:  code/._SnakefilePAS
    Untracked:  code/._SnakefilefiltPAS
    Untracked:  code/._aAPAqtl_nominal39ind.sh
    Untracked:  code/._apaQTLCorrectPvalMakeQQ.R
    Untracked:  code/._apaQTL_Nominal.sh
    Untracked:  code/._apaQTL_permuted.sh
    Untracked:  code/._bed2saf.py
    Untracked:  code/._callPeaksYL.py
    Untracked:  code/._chooseAnno2SAF.py
    Untracked:  code/._chooseSignalSite
    Untracked:  code/._chooseSignalSite.py
    Untracked:  code/._cluster.json
    Untracked:  code/._clusterPAS.json
    Untracked:  code/._clusterfiltPAS.json
    Untracked:  code/._config.yaml
    Untracked:  code/._config2.yaml
    Untracked:  code/._configOLD.yaml
    Untracked:  code/._convertNumeric.py
    Untracked:  code/._dag.pdf
    Untracked:  code/._extractGenotypes.py
    Untracked:  code/._filter5perc.R
    Untracked:  code/._filter5percPheno.py
    Untracked:  code/._filterpeaks.py
    Untracked:  code/._fixFChead.py
    Untracked:  code/._make5percPeakbed.py
    Untracked:  code/._makeFileID.py
    Untracked:  code/._makePheno.py
    Untracked:  code/._mergeAllBam.sh
    Untracked:  code/._mergeByFracBam.sh
    Untracked:  code/._mergePeaks.sh
    Untracked:  code/._namePeaks.py
    Untracked:  code/._peak2PAS.py
    Untracked:  code/._peakFC.sh
    Untracked:  code/._pheno2countonly.R
    Untracked:  code/._quantassign2parsedpeak.py
    Untracked:  code/._selectNominalPvalues.py
    Untracked:  code/._snakemakePAS.batch
    Untracked:  code/._snakemakefiltPAS.batch
    Untracked:  code/._submit-snakemakePAS.sh
    Untracked:  code/._submit-snakemakefiltPAS.sh
    Untracked:  code/.snakemake/
    Untracked:  code/APAqtl_nominal.err
    Untracked:  code/APAqtl_nominal.out
    Untracked:  code/APAqtl_nominal_39.err
    Untracked:  code/APAqtl_nominal_39.out
    Untracked:  code/APAqtl_permuted.err
    Untracked:  code/APAqtl_permuted.out
    Untracked:  code/BothFracDTPlotGeneRegions.err
    Untracked:  code/BothFracDTPlotGeneRegions.out
    Untracked:  code/DistPAS2Sig.py
    Untracked:  code/README.md
    Untracked:  code/Rplots.pdf
    Untracked:  code/Upstream100Bases_general.py
    Untracked:  code/aAPAqtl_nominal39ind.sh
    Untracked:  code/bam2bw.err
    Untracked:  code/bam2bw.out
    Untracked:  code/dag.pdf
    Untracked:  code/dagPAS.pdf
    Untracked:  code/dagfiltPAS.pdf
    Untracked:  code/extractGenotypes.py
    Untracked:  code/findbuginpeaks.R
    Untracked:  code/get100upPAS.py
    Untracked:  code/getSeq100up.sh
    Untracked:  code/getseq100up.err
    Untracked:  code/getseq100up.out
    Untracked:  code/log/
    Untracked:  code/run_DistPAS2Sig.err
    Untracked:  code/run_DistPAS2Sig.out
    Untracked:  code/run_distPAS2Sig.sh
    Untracked:  code/selectNominalPvalues.py
    Untracked:  code/snakePASlog.out
    Untracked:  code/snakefiltPASlog.out
    Untracked:  data/DTmatrix/
    Untracked:  data/PAS/
    Untracked:  data/QTLGenotypes/
    Untracked:  data/README.md
    Untracked:  data/SignalSiteFiles/
    Untracked:  data/ThirtyNineIndQtl_nominal/
    Untracked:  data/apaQTLNominal/
    Untracked:  data/apaQTLPermuted/
    Untracked:  data/apaQTLs/
    Untracked:  data/assignedPeaks/
    Untracked:  data/bam/
    Untracked:  data/bam_clean/
    Untracked:  data/bam_waspfilt/
    Untracked:  data/bed_10up/
    Untracked:  data/bed_clean/
    Untracked:  data/bed_clean_sort/
    Untracked:  data/bed_waspfilter/
    Untracked:  data/bedsort_waspfilter/
    Untracked:  data/exampleQTLs/
    Untracked:  data/fastq/
    Untracked:  data/filterPeaks/
    Untracked:  data/inclusivePeaks/
    Untracked:  data/inclusivePeaks_FC/
    Untracked:  data/mergedBG/
    Untracked:  data/mergedBW_byfrac/
    Untracked:  data/mergedBam/
    Untracked:  data/mergedbyFracBam/
    Untracked:  data/nuc_10up/
    Untracked:  data/nuc_10upclean/
    Untracked:  data/peakCoverage/
    Untracked:  data/peaks_5perc/
    Untracked:  data/phenotype/
    Untracked:  data/phenotype_5perc/
    Untracked:  data/sort/
    Untracked:  data/sort_clean/
    Untracked:  data/sort_waspfilter/
    Untracked:  nohup.out
    Untracked:  output/._.DS_Store
    Untracked:  output/dtPlots/
    Untracked:  output/fastqc/

Unstaged changes:
    Modified:   analysis/PASusageQC.Rmd
    Modified:   analysis/corrbetweenind.Rmd
    Deleted:    code/Upstream10Bases_general.py
    Modified:   code/apaQTLCorrectPvalMakeQQ.R
    Modified:   code/apaQTL_permuted.sh
    Modified:   code/bed2saf.py
    Deleted:    code/test.txt

</code></pre>
<p>
Note that any generated files, e.g.Â HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If youâ€™ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/0f39c9b26884cd39b2a6647f77e5969736dc3128/analysis/NewvOldPeakQC.Rmd" target="_blank">0f39c9b</a>
</td>
<td>
brimittleman
</td>
<td>
2019-05-02
</td>
<td>
add usage in old data
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/apaQTL/3d56b3b5df8f22c7d45b750196f6d9bdaba83db8/docs/NewvOldPeakQC.html" target="_blank">3d56b3b</a>
</td>
<td>
brimittleman
</td>
<td>
2019-05-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/e9021d256562149996342ed6ed485537c3c4cbb3/analysis/NewvOldPeakQC.Rmd" target="_blank">e9021d2</a>
</td>
<td>
brimittleman
</td>
<td>
2019-05-01
</td>
<td>
compare usage in new, by ind set
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/apaQTL/09090d7834800361bafe3b2d16b2b9569c44517f/docs/NewvOldPeakQC.html" target="_blank">09090d7</a>
</td>
<td>
brimittleman
</td>
<td>
2019-05-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/f4f5b21553b16c8948f83b538061dab32ad4d09e/analysis/NewvOldPeakQC.Rmd" target="_blank">f4f5b21</a>
</td>
<td>
brimittleman
</td>
<td>
2019-05-01
</td>
<td>
add number of peak per gene analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/apaQTL/03ea0f7f10cb119174b2b47cfbe388beb5c34178/docs/NewvOldPeakQC.html" target="_blank">03ea0f7</a>
</td>
<td>
brimittleman
</td>
<td>
2019-05-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/f58797082799d020c419ca0d45c1dc1f5b91d71f/analysis/NewvOldPeakQC.Rmd" target="_blank">f587970</a>
</td>
<td>
brimittleman
</td>
<td>
2019-05-01
</td>
<td>
fix join
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/apaQTL/2ec0f07b8a52a3e281551ecd05d0c8d327545e7c/docs/NewvOldPeakQC.html" target="_blank">2ec0f07</a>
</td>
<td>
brimittleman
</td>
<td>
2019-05-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/3153f49ecd4c0b6cd0e0eb18401c762f33700c6b/analysis/NewvOldPeakQC.Rmd" target="_blank">3153f49</a>
</td>
<td>
brimittleman
</td>
<td>
2019-05-01
</td>
<td>
old hists
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/apaQTL/15e195f787e3347dcffd6a0c82574e29700290a8/docs/NewvOldPeakQC.html" target="_blank">15e195f</a>
</td>
<td>
brimittleman
</td>
<td>
2019-05-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/f5e6fd66e6d8d41ee88c200a55cb39e2d0494d66/analysis/NewvOldPeakQC.Rmd" target="_blank">f5e6fd6</a>
</td>
<td>
brimittleman
</td>
<td>
2019-05-01
</td>
<td>
add random 15 ind hists
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/apaQTL/9ccb06ecf89bdb9b56a969d5b72ccdca8249d30b/docs/NewvOldPeakQC.html" target="_blank">9ccb06e</a>
</td>
<td>
brimittleman
</td>
<td>
2019-04-30
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/eb0bd9570e8b3d8e78a72d51a70c388f0f8e267d/analysis/NewvOldPeakQC.Rmd" target="_blank">eb0bd95</a>
</td>
<td>
brimittleman
</td>
<td>
2019-04-30
</td>
<td>
add write out step
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/brimittleman/apaQTL/ac656aad301db960e1e2b218a7235b295395057d/docs/NewvOldPeakQC.html" target="_blank">ac656aa</a>
</td>
<td>
brimittleman
</td>
<td>
2019-04-30
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/f9b8195e7d9465dff3beb1cb72b790d740f2e1d0/analysis/NewvOldPeakQC.Rmd" target="_blank">f9b8195</a>
</td>
<td>
brimittleman
</td>
<td>
2019-04-30
</td>
<td>
understand usage of new pas
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(reshape2)
library(tidyverse)</code></pre>
<pre><code>â”€â”€ Attaching packages â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€ tidyverse 1.2.1 â”€â”€</code></pre>
<pre><code>âœ” ggplot2 3.1.0       âœ” purrr   0.3.2  
âœ” tibble  2.1.1       âœ” dplyr   0.8.0.1
âœ” tidyr   0.8.3       âœ” stringr 1.3.1  
âœ” readr   1.3.1       âœ” forcats 0.3.0  </code></pre>
<pre><code>â”€â”€ Conflicts â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€ tidyverse_conflicts() â”€â”€
âœ– dplyr::filter() masks stats::filter()
âœ– dplyr::lag()    masks stats::lag()</code></pre>
<pre class="r"><code>library(workflowr)</code></pre>
<pre><code>This is workflowr version 1.3.0
Run ?workflowr for help getting started</code></pre>
<p>These results have 30k more PAS than the previous runs. I also see a confusing shift in mean usage for all of the PAS. I want to compare the distribution of usage for different sets of individuals to see if there is something inherently different about the 15 new individuals.</p>
<div id="new-vs-old-peaks" class="section level2">
<h2>New vs old peaks</h2>
<p>I want to compare the usage of the new peaks compared to the overall mean usage. To do this I need to seperate the new and old PAS.</p>
<pre class="r"><code>newPAS5perc=read.table(&quot;../data/PAS/APAPAS_GeneLocAnno.5perc.bed&quot;, stringsAsFactors = F, col.names = c(&quot;chr&quot;, &quot;start&quot;,&quot;end&quot;, &quot;ID&quot;, &quot;score&quot;, &quot;strand&quot;))
oldPAS5perc=read.table(&quot;../../threeprimeseq/data/peaks4DT/APAPAS_5percCov_fixedStrand.bed&quot;, stringsAsFactors = F, col.names = c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;ID&quot;, &quot;score&quot;, &quot;strand&quot;))

uniqnew=newPAS5perc %&gt;% anti_join(oldPAS5perc, by=c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;))</code></pre>
<p>Pull in the usage of the peaks:</p>
<div id="total" class="section level3">
<h3>Total</h3>
<pre class="r"><code>totalPeakUs=read.table(&quot;../data/phenotype/APApeak_Phenotype_GeneLocAnno.Total.fc&quot;, header = T, stringsAsFactors = F) %&gt;% separate(chrom, sep = &quot;:&quot;, into = c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;id&quot;)) %&gt;% separate(id, sep=&quot;_&quot;, into=c(&quot;gene&quot;, &quot;loc&quot;, &quot;strand&quot;, &quot;peak&quot;))</code></pre>
<pre><code>Warning: Expected 4 pieces. Additional pieces discarded in 4 rows [14735,
14736, 14737, 14738].</code></pre>
<pre class="r"><code>ind=colnames(totalPeakUs)[8:dim(totalPeakUs)[2]]
totalPeakUs_CountNum=read.table(&quot;../data/phenotype/APApeak_Phenotype_GeneLocAnno.Total.CountsOnlyNumeric&quot;, col.names = ind)


#numeric with anno
totalPeak=as.data.frame(cbind(totalPeakUs[,1:7], totalPeakUs_CountNum))

totalPeakUs_CountNum_mean=rowMeans(totalPeakUs_CountNum)

#append mean to anno
TotalPeakUSMean=as.data.frame(cbind(totalPeakUs[,1:7],mean=totalPeakUs_CountNum_mean))</code></pre>
<pre class="r"><code>uniqnewPasnum=uniqnew  %&gt;% separate(ID ,into=c(&quot;peaknum&quot;, &quot;geneloc&quot;),sep=&quot;:&quot;) %&gt;% mutate(peak=paste(&quot;peak&quot;, peaknum, sep=&quot;&quot;)) %&gt;% select(peak)</code></pre>
<p>Filter these inthe mean usage:</p>
<pre class="r"><code>TotalPeakUSMeanClass= TotalPeakUSMean %&gt;% mutate(New=ifelse(peak %in% uniqnewPasnum$peak,&quot;new&quot;, &quot;original&quot;)) %&gt;% mutate(Cutoff=ifelse(mean&gt;=.05, &quot;Yes&quot;, &quot;No&quot;))

mean(TotalPeakUSMean$mean)</code></pre>
<pre><code>[1] 0.2378282</code></pre>
<p>Plot:</p>
<pre class="r"><code>ggplot(TotalPeakUSMeanClass, aes(y=mean,x=New)) + geom_violin() + geom_hline(yintercept = mean(TotalPeakUSMean$mean), col=&quot;red&quot;) +  geom_hline(yintercept = .05, col=&quot;Blue&quot;)</code></pre>
<p><img src="figure/NewvOldPeakQC.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-1">
Past versions of unnamed-chunk-6-1.png
</button>
</p>
<div id="fig-unnamed-chunk-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/09090d7834800361bafe3b2d16b2b9569c44517f/docs/figure/NewvOldPeakQC.Rmd/unnamed-chunk-6-1.png" target="_blank">09090d7</a>
</td>
<td>
brimittleman
</td>
<td>
2019-05-01
</td>
</tr>
<tr>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/03ea0f7f10cb119174b2b47cfbe388beb5c34178/docs/figure/NewvOldPeakQC.Rmd/unnamed-chunk-6-1.png" target="_blank">03ea0f7</a>
</td>
<td>
brimittleman
</td>
<td>
2019-05-01
</td>
</tr>
<tr>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/ac656aad301db960e1e2b218a7235b295395057d/docs/figure/NewvOldPeakQC.Rmd/unnamed-chunk-6-1.png" target="_blank">ac656aa</a>
</td>
<td>
brimittleman
</td>
<td>
2019-04-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>TotalPeakUSMeanClass_newonly= TotalPeakUSMeanClass %&gt;% filter(New==&quot;new&quot;)

ggplot(TotalPeakUSMeanClass_newonly, aes(y=mean, x=&quot;&quot;)) + geom_violin() + geom_hline(yintercept = mean(TotalPeakUSMean$mean), col=&quot;red&quot;) +  geom_hline(yintercept = .05, col=&quot;Blue&quot;) + labs(x=&quot;&quot;, y=&quot;Mean Usage&quot;, title=&quot;Mean Usage of New PAS&quot;)</code></pre>
<p><img src="figure/NewvOldPeakQC.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/09090d7834800361bafe3b2d16b2b9569c44517f/docs/figure/NewvOldPeakQC.Rmd/unnamed-chunk-7-1.png" target="_blank">09090d7</a>
</td>
<td>
brimittleman
</td>
<td>
2019-05-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(TotalPeakUSMeanClass_newonly, aes(x=mean)) + geom_histogram(bins=50) + geom_vline(xintercept = mean(TotalPeakUSMean$mean), col=&quot;red&quot;)</code></pre>
<p><img src="figure/NewvOldPeakQC.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-1">
Past versions of unnamed-chunk-8-1.png
</button>
</p>
<div id="fig-unnamed-chunk-8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/09090d7834800361bafe3b2d16b2b9569c44517f/docs/figure/NewvOldPeakQC.Rmd/unnamed-chunk-8-1.png" target="_blank">09090d7</a>
</td>
<td>
brimittleman
</td>
<td>
2019-05-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>This shows me the new peaks are the peaks that barely passed the cuttoff before.</p>
<p>write out file with information about new and old peaks</p>
<pre class="r"><code>Peak_newOld=TotalPeakUSMeanClass %&gt;% select(-mean)
write.table(Peak_newOld, file=&quot;../data/peaks_5perc/NewVOldPeaks.txt&quot;, col.names = T, row.names = F, quote=F)</code></pre>
</div>
</div>
<div id="distirubution-in-different-sets-of-15-ind" class="section level2">
<h2>Distirubution in different sets of 15 ind</h2>
<p>I want to see if the new 15 individuals are driving the change in the peak mean distribution. I want to make a function that take a vector of individuals, filters the usage dataframe and plots the histogram.</p>
<div id="total-1" class="section level3">
<h3>Total:</h3>
<p>First I will upload the usage dataframe.</p>
<pre class="r"><code>totCounts=read.table(&quot;../data/phenotype/APApeak_Phenotype_GeneLocAnno.Total.fc&quot;, stringsAsFactors = F, header = T)
ind=colnames(totCounts)[2:55]
totUsage=read.table(&quot;../data/phenotype/APApeak_Phenotype_GeneLocAnno.Total.CountsOnlyNumeric&quot;, stringsAsFactors = F, header = F,col.names = ind)



batch1.2.3=read.table(&quot;../data/MetaDataSequencing.txt&quot;, header=T,stringsAsFactors = F)%&gt;% filter(fraction==&quot;total&quot;) %&gt;%  select(line, batch) %&gt;% filter(batch != 4)
oldind=batch1.2.3$line

batch4=read.table(&quot;../data/MetaDataSequencing.txt&quot;, header=T,stringsAsFactors = F)%&gt;% filter(fraction==&quot;total&quot;) %&gt;%  select(line, batch) %&gt;% filter(batch == 4)

newInd=batch4$line</code></pre>
<pre class="r"><code>UsageHist= function(indVec,title,totUsage=totUsage){
  totUsage_ind=totUsage %&gt;% select(indVec)
  meanVec=rowMeans(totUsage_ind)
  hist(meanVec, main=title,xlab=&quot;Mean Usage&quot;)
}</code></pre>
<p>RUn this for different itterations of individuals:</p>
<p>Pick 15 random individuals from old:</p>
<pre class="r"><code>sampl1=sample(oldind, 15)
sampl2=sample(oldind, 15)
sampl3=sample(oldind, 15)
sampl4=sample(oldind, 15)

png(&quot;../output/newtot.png&quot;)
par(mfrow=c(3,2))
UsageHist(indVec=newInd,title=&quot;Total Usage New (15ind)&quot;,totUsage=totUsage)
UsageHist(indVec=oldind,title=&quot;Total Usage Old (39 ind)&quot;,totUsage=totUsage)
UsageHist(indVec=sampl1,title=&quot;Total Usage Sample 15 Old&quot;,totUsage=totUsage)
UsageHist(indVec=sampl2,title=&quot;Total Usage Sample 15 Old&quot;,totUsage=totUsage)
UsageHist(indVec=sampl3,title=&quot;Total Usage Sample 15 Old&quot;,totUsage=totUsage)
UsageHist(indVec=sampl4,title=&quot;Total Usage Sample 15 Old&quot;,totUsage=totUsage)
dev.off()</code></pre>
<pre><code>png 
  2 </code></pre>
</div>
<div id="nuclear" class="section level3">
<h3>Nuclear</h3>
<pre class="r"><code>nucCounts=read.table(&quot;../data/phenotype/APApeak_Phenotype_GeneLocAnno.Nuclear.fc&quot;, stringsAsFactors = F, header = T)
ind=colnames(nucCounts)[2:55]
nucUsage=read.table(&quot;../data/phenotype/APApeak_Phenotype_GeneLocAnno.Nuclear.CountsOnlyNumeric&quot;, stringsAsFactors = F, header = F,col.names = ind)</code></pre>
<pre class="r"><code>UsageHist_nuc= function(indVec,title,nucUsage=nucUsage){
  nucUsage_ind=nucUsage %&gt;% select(indVec)
  meanVec=rowMeans(nucUsage_ind)
  hist(meanVec, main=title,xlab=&quot;Mean Usage&quot;)
}</code></pre>
<pre class="r"><code>png(&quot;../output/newnuc.png&quot;)
par(mfrow=c(3,2))
UsageHist_nuc(indVec=newInd,title=&quot;Nuclear Usage New (15ind)&quot;,nucUsage=nucUsage)
UsageHist_nuc(indVec=oldind,title=&quot;Nuclear Usage Old (39ind)&quot;,nucUsage=nucUsage)
UsageHist_nuc(indVec=sampl1,title=&quot;Nuclear Usage Sample 15 Old&quot;,nucUsage=nucUsage)
UsageHist_nuc(indVec=sampl2,title=&quot;Nuclear Usage Sample 15 Old&quot;,nucUsage=nucUsage)
UsageHist_nuc(indVec=sampl3,title=&quot;Nuclear Usage Sample 15 Old&quot;,nucUsage=nucUsage)
UsageHist_nuc(indVec=sampl4,title=&quot;Nuclear Usage Sample 15 Old&quot;,nucUsage=nucUsage)
dev.off()</code></pre>
<pre><code>png 
  2 </code></pre>
<p>Old</p>
<pre class="r"><code>oldtotalCount=read.table(&quot;../../threeprimeseq/data/phenotypes_filtPeakTranscript_noMP_GeneLocAnno/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Total.fixed.pheno.fc&quot;,header=T,stringsAsFactors = F)
indOld=colnames(oldtotalCount)[2:56]
oldtotalUsage=read.table(&quot;../../threeprimeseq/data/phenotypes_filtPeakTranscript_noMP_GeneLocAnno/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Total.fixed.pheno.CountsOnlyNumeric.txt&quot;,col.names = indOld,stringsAsFactors = F)</code></pre>
<pre class="r"><code>png(&quot;../output/oldtot.png&quot;)
par(mfrow=c(3,2))
UsageHist(indVec=newInd,title=&quot;Old total Usage (15ind)&quot;,totUsage=oldtotalUsage)
UsageHist(indVec=oldind,title=&quot;Old total Usage (39ind)&quot;,totUsage=oldtotalUsage)
UsageHist(indVec=sampl1,title=&quot;Old total Usage sample 15 ind&quot;,totUsage=oldtotalUsage)
UsageHist(indVec=sampl2,title=&quot;Old total Usage sample 15 ind&quot;,totUsage=oldtotalUsage)
UsageHist(indVec=sampl3,title=&quot;Old total Usage sample 15 ind&quot;,totUsage=oldtotalUsage)
UsageHist(indVec=sampl4,title=&quot;Old total Usage sample 15 ind&quot;,totUsage=oldtotalUsage)
dev.off()</code></pre>
<pre><code>png 
  2 </code></pre>
<pre class="r"><code>oldnuclearCount=read.table(&quot;../../threeprimeseq/data/phenotypes_filtPeakTranscript_noMP_GeneLocAnno/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Nuclear.fixed.pheno.fc&quot;,header=T,stringsAsFactors = F)
indOldN=colnames(oldnuclearCount)[2:56]
oldnuclearUsage=read.table(&quot;../../threeprimeseq/data/phenotypes_filtPeakTranscript_noMP_GeneLocAnno/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno_NoMP_sm_quant.Nuclear.fixed.pheno.CountsOnlyNumeric.txt&quot;,col.names = indOldN,stringsAsFactors = F)</code></pre>
<pre class="r"><code>png(&quot;../output/oldnuc.png&quot;)
par(mfrow=c(3,2))
UsageHist_nuc(indVec=newInd,title=&quot;Old nuclear Usage (15ind)&quot;,nucUsage=oldnuclearUsage)
UsageHist_nuc(indVec=oldind,title=&quot;Old nuclear Usage (39ind)&quot;,nucUsage=oldnuclearUsage)
UsageHist_nuc(indVec=sampl1,title=&quot;Old nuclear Usage sample 15 ind&quot;,nucUsage=oldnuclearUsage)
UsageHist_nuc(indVec=sampl2,title=&quot;Old nuclear Usage sample 15 ind&quot;,nucUsage=oldnuclearUsage)
UsageHist_nuc(indVec=sampl3,title=&quot;Old nuclear Usage sample 15 ind&quot;,nucUsage=oldnuclearUsage)
UsageHist_nuc(indVec=sampl4,title=&quot;Old nuclear Usage sample 15 ind&quot;,nucUsage=oldnuclearUsage)
dev.off()</code></pre>
<pre><code>png 
  2 </code></pre>
</div>
</div>
<div id="usage-of-new-peaks-in-old-data" class="section level2">
<h2>Usage of new peaks in old data</h2>
<p>I have the old usage. I want to filter the new peaks from this:</p>
<pre class="r"><code>oldtotusage_anno=as.data.frame(cbind(chrom=oldtotalCount$chrom,oldtotalUsage )) %&gt;% separate(chrom, into=c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;peakID&quot;), sep=&quot;:&quot;)

TotalPeakUSMeanClass_newonly=TotalPeakUSMeanClass %&gt;% filter(New==&quot;new&quot;)

oldtotusage_anno_new=oldtotusage_anno %&gt;% semi_join(TotalPeakUSMeanClass_newonly, by=c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;)) %&gt;% select(-chr, -start,-end,-peakID)


oldtotusage_anno_new_mean=rowMeans(oldtotusage_anno_new)

plot(oldtotusage_anno_new_mean, main=&quot;Total Usage of new peaks in old data&quot;, ylab=&quot;old usage means Percent&quot;)</code></pre>
<p><img src="figure/NewvOldPeakQC.Rmd/unnamed-chunk-20-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-20-1">
Past versions of unnamed-chunk-20-1.png
</button>
</p>
<div id="fig-unnamed-chunk-20-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/09090d7834800361bafe3b2d16b2b9569c44517f/docs/figure/NewvOldPeakQC.Rmd/unnamed-chunk-20-1.png" target="_blank">09090d7</a>
</td>
<td>
brimittleman
</td>
<td>
2019-05-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>length(oldtotusage_anno_new_mean)</code></pre>
<pre><code>[1] 14642</code></pre>
<p>This shows me that we called 14 thousand of the new peaks in the old set but they were all super low coverage.</p>
<p>Where are the new peaks, genes with a lot of peaks? or genes with less peaks</p>
<pre class="r"><code>newPAS5perc_pergene=newPAS5perc %&gt;% separate(ID, into=c(&quot;peaknum&quot;, &quot;geneID&quot;), sep=&quot;:&quot;) %&gt;% separate(geneID, into=c(&quot;gene&quot;, &quot;loc&quot;),sep=&quot;_&quot;) %&gt;% group_by(gene) %&gt;% summarise(nPeak=n())

ggplot(newPAS5perc_pergene,aes(x=nPeak)) + geom_histogram(bins=100)</code></pre>
<p><img src="figure/NewvOldPeakQC.Rmd/unnamed-chunk-21-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-21-1">
Past versions of unnamed-chunk-21-1.png
</button>
</p>
<div id="fig-unnamed-chunk-21-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/09090d7834800361bafe3b2d16b2b9569c44517f/docs/figure/NewvOldPeakQC.Rmd/unnamed-chunk-21-1.png" target="_blank">09090d7</a>
</td>
<td>
brimittleman
</td>
<td>
2019-05-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>nrow(newPAS5perc_pergene)</code></pre>
<pre><code>[1] 15456</code></pre>
<p>Look at which genes the new peaks are in.</p>
<pre class="r"><code>uniqnew_genes=uniqnew %&gt;% separate(ID, into=c(&quot;peaknum&quot;, &quot;geneID&quot;), sep=&quot;:&quot;) %&gt;% separate(geneID, into=c(&quot;gene&quot;, &quot;loc&quot;),sep=&quot;_&quot;) %&gt;% group_by(gene) %&gt;% summarise(npeakadded=n())


ggplot(uniqnew_genes, aes(x=npeakadded)) + geom_histogram(bins=100) + labs(title=&quot;Number of peaks added per gene\n(added peak in 10827 genes of 15456 genes)&quot;, x=&quot;Number of Peaks&quot;, y=&quot;Number of Genes&quot;)</code></pre>
<p><img src="figure/NewvOldPeakQC.Rmd/unnamed-chunk-22-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-22-1">
Past versions of unnamed-chunk-22-1.png
</button>
</p>
<div id="fig-unnamed-chunk-22-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/09090d7834800361bafe3b2d16b2b9569c44517f/docs/figure/NewvOldPeakQC.Rmd/unnamed-chunk-22-1.png" target="_blank">09090d7</a>
</td>
<td>
brimittleman
</td>
<td>
2019-05-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Look at these genes compared to distribution for number of peaks in all peaks before filter</p>
<pre class="r"><code>allPAS=read.table(&quot;../data/assignedPeaks/APApeaks.ALLChrom.Filtered.Named.GeneLocAnnoPARSED.SAF&quot;, header = T, stringsAsFactors = F) %&gt;% separate(GeneID, into = c(&quot;peak&quot;, &quot;Chrom&quot;, &quot;Peakstart&quot;, &quot;PeakEnd&quot;, &quot;strand&quot;, &quot;geneid&quot;),sep=&quot;:&quot;) %&gt;% separate(geneid, into=c(&quot;gene&quot;, &quot;loc&quot;), sep=&quot;_&quot;) %&gt;% group_by(gene) %&gt;% summarise(nPeakAll=n()) %&gt;% mutate(AddedPeak=ifelse(gene %in% uniqnew_genes$gene, &quot;yes&quot;, &quot;no&quot;))</code></pre>
<pre><code>Warning: Expected 2 pieces. Additional pieces discarded in 4 rows [14735,
14736, 14737, 14738].</code></pre>
<pre class="r"><code>ggplot(allPAS, aes(x=nPeakAll))+ geom_histogram(bins = 100) + facet_grid(~AddedPeak)</code></pre>
<p><img src="figure/NewvOldPeakQC.Rmd/unnamed-chunk-23-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-23-1">
Past versions of unnamed-chunk-23-1.png
</button>
</p>
<div id="fig-unnamed-chunk-23-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/09090d7834800361bafe3b2d16b2b9569c44517f/docs/figure/NewvOldPeakQC.Rmd/unnamed-chunk-23-1.png" target="_blank">09090d7</a>
</td>
<td>
brimittleman
</td>
<td>
2019-05-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Mean number of peaks for genes where we added, this could help us understand the distribution shift:</p>
<pre class="r"><code>allPAS_withnewpas=allPAS %&gt;% filter(AddedPeak==&quot;yes&quot;)

summary(allPAS_withnewpas$nPeakAll)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
  1.000   3.000   6.000   9.725  12.000 201.000 </code></pre>
<p>Look at n peaks per gene in old data and npeaks per gene in new- join and subtract</p>
<pre class="r"><code>#newPAS5perc_pergene

oldPAS5perc_pergene= oldPAS5perc%&gt;% separate(ID, into=c(&quot;gene&quot;, &quot;peaknum&quot;), sep=&quot;:&quot;) %&gt;% group_by(gene) %&gt;% summarise(nPeakOld=n())

nrow(oldPAS5perc_pergene)</code></pre>
<pre><code>[1] 15219</code></pre>
<p>Join:</p>
<pre class="r"><code>npeakpergenebot=oldPAS5perc_pergene %&gt;% full_join(newPAS5perc_pergene, by=&quot;gene&quot;) %&gt;% replace_na(list(nPeakOld = 0, nPeak = 0)) %&gt;% mutate(NewMinOld=nPeak-nPeakOld)


ggplot(npeakpergenebot, aes(x=nPeakOld, y=nPeak)) + geom_point() + geom_smooth(method=&quot;lm&quot;) + annotate(&quot;text&quot;, label=&quot;r2=0.4447&quot;, x=12, y=2) + labs(title=&quot;Number of Peaks in old vs new data&quot;,y=&quot;Number of peaks new data&quot;, x=&quot;Number of peaks old data&quot;)</code></pre>
<p><img src="figure/NewvOldPeakQC.Rmd/unnamed-chunk-26-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-26-1">
Past versions of unnamed-chunk-26-1.png
</button>
</p>
<div id="fig-unnamed-chunk-26-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/09090d7834800361bafe3b2d16b2b9569c44517f/docs/figure/NewvOldPeakQC.Rmd/unnamed-chunk-26-1.png" target="_blank">09090d7</a>
</td>
<td>
brimittleman
</td>
<td>
2019-05-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Correlation between number of peaks before and after:</p>
<pre class="r"><code>summary(lm(npeakpergenebot$nPeak~npeakpergenebot$nPeakOld))</code></pre>
<pre><code>
Call:
lm(formula = npeakpergenebot$nPeak ~ npeakpergenebot$nPeakOld)

Residuals:
     Min       1Q   Median       3Q      Max 
-10.1193  -1.6746  -1.3302   0.9475  17.3254 

Coefficients:
                         Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)               0.98570    0.04322    22.8   &lt;2e-16 ***
npeakpergenebot$nPeakOld  1.34447    0.01201   112.0   &lt;2e-16 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 3.185 on 15650 degrees of freedom
Multiple R-squared:  0.4448,    Adjusted R-squared:  0.4447 
F-statistic: 1.254e+04 on 1 and 15650 DF,  p-value: &lt; 2.2e-16</code></pre>
<pre class="r"><code>summary(npeakpergenebot$NewMinOld)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
 -6.000   0.000   1.000   1.988   3.000  20.000 </code></pre>
<pre class="r"><code>ggplot(npeakpergenebot, aes(x=NewMinOld)) + geom_histogram(bins=100)</code></pre>
<p><img src="figure/NewvOldPeakQC.Rmd/unnamed-chunk-28-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-28-1">
Past versions of unnamed-chunk-28-1.png
</button>
</p>
<div id="fig-unnamed-chunk-28-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/09090d7834800361bafe3b2d16b2b9569c44517f/docs/figure/NewvOldPeakQC.Rmd/unnamed-chunk-28-1.png" target="_blank">09090d7</a>
</td>
<td>
brimittleman
</td>
<td>
2019-05-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="usage-of-new-peaks-by-ind-set" class="section level2">
<h2>Usage of new peaks by ind set</h2>
<div id="total-2" class="section level3">
<h3>Total</h3>
<pre class="r"><code>totalPeak_inNew=totalPeak %&gt;% filter(peak %in% uniqnewPasnum$peak)

totalPeak_inNew_melt=melt(totalPeak_inNew,id.vars=c(&quot;chr&quot;, &quot;start&quot;,&quot;end&quot;, &quot;gene&quot;,&quot;loc&quot;, &quot;peak&quot;,&quot;strand&quot;),value.name = &quot;Usage&quot; ,variable.name = &quot;Ind&quot;) %&gt;% mutate(New15=ifelse(Ind %in% batch4$line, &quot;Yes&quot;, &quot;No&quot;))


ggplot(totalPeak_inNew_melt, aes(x=New15, y=Usage, fill=New15))+geom_boxplot(width=.5) + theme(legend.position = &quot;none&quot;) + labs(x=&quot;Individuals in New batch&quot;)</code></pre>
<p><img src="figure/NewvOldPeakQC.Rmd/unnamed-chunk-29-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-29-1">
Past versions of unnamed-chunk-29-1.png
</button>
</p>
<div id="fig-unnamed-chunk-29-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/3d56b3b5df8f22c7d45b750196f6d9bdaba83db8/docs/figure/NewvOldPeakQC.Rmd/unnamed-chunk-29-1.png" target="_blank">3d56b3b</a>
</td>
<td>
brimittleman
</td>
<td>
2019-05-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Look if means are different:</p>
<pre class="r"><code>totalPeak_inNew_meltGroup=totalPeak_inNew_melt %&gt;% group_by(New15,peak) %&gt;% summarise(meanUsage=mean(Usage))

ggplot(totalPeak_inNew_meltGroup, aes(x=New15, y=meanUsage, fill=New15))+geom_boxplot(width=.5) + theme(legend.position = &quot;none&quot;) + labs(x=&quot;Individuals in New batch&quot;, y=&quot;Mean Usage in Group&quot;, title=&quot;Mean usage for new PAS in 39 ind v new 15&quot;)</code></pre>
<p><img src="figure/NewvOldPeakQC.Rmd/unnamed-chunk-30-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-30-1">
Past versions of unnamed-chunk-30-1.png
</button>
</p>
<div id="fig-unnamed-chunk-30-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/3d56b3b5df8f22c7d45b750196f6d9bdaba83db8/docs/figure/NewvOldPeakQC.Rmd/unnamed-chunk-30-1.png" target="_blank">3d56b3b</a>
</td>
<td>
brimittleman
</td>
<td>
2019-05-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(totalPeak_inNew_meltGroup, aes(x=meanUsage, group=New15, fill=New15))+geom_density(alpha=.3)  + labs( x=&quot;Mean Usage in Group&quot;, title=&quot;Mean usage for new PAS in 39 ind v new 15&quot;)+ scale_fill_discrete(name = &quot;Ind in new 15&quot;)</code></pre>
<p><img src="figure/NewvOldPeakQC.Rmd/unnamed-chunk-31-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-31-1">
Past versions of unnamed-chunk-31-1.png
</button>
</p>
<div id="fig-unnamed-chunk-31-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/brimittleman/apaQTL/blob/3d56b3b5df8f22c7d45b750196f6d9bdaba83db8/docs/figure/NewvOldPeakQC.Rmd/unnamed-chunk-31-1.png" target="_blank">3d56b3b</a>
</td>
<td>
brimittleman
</td>
<td>
2019-05-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>totalPeak_inNew_meltGroup_spre=totalPeak_inNew_meltGroup %&gt;% spread(New15,meanUsage)


ggplot(totalPeak_inNew_meltGroup_spre,aes(x=No, y=Yes)) + geom_point() + geom_density_2d() + geom_smooth(method = &quot;lm&quot;)+annotate(&quot;text&quot;, label=&quot;New=old+0.053680&quot; ,x=.8,y=0) + labs(x=&quot;Original 39 Mean&quot;, y=&quot;Original 15 Mean&quot;, title=&quot;Mean Usage in New data by Ind set&quot;)</code></pre>
<p><img src="figure/NewvOldPeakQC.Rmd/unnamed-chunk-32-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>summary(lm(data=totalPeak_inNew_meltGroup_spre, Yes~No))</code></pre>
<pre><code>
Call:
lm(formula = Yes ~ No, data = totalPeak_inNew_meltGroup_spre)

Residuals:
     Min       1Q   Median       3Q      Max 
-0.80920 -0.08513 -0.03408  0.05478  0.86924 

Coefficients:
            Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept) 0.053680   0.001055   50.86   &lt;2e-16 ***
No          1.002046   0.003416  293.35   &lt;2e-16 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 0.1584 on 42931 degrees of freedom
Multiple R-squared:  0.6672,    Adjusted R-squared:  0.6671 
F-statistic: 8.605e+04 on 1 and 42931 DF,  p-value: &lt; 2.2e-16</code></pre>
</div>
<div id="nuclear-1" class="section level3">
<h3>Nuclear</h3>
<pre class="r"><code>nuclearPeakUs=read.table(&quot;../data/phenotype/APApeak_Phenotype_GeneLocAnno.Nuclear.fc&quot;, header = T, stringsAsFactors = F) %&gt;% separate(chrom, sep = &quot;:&quot;, into = c(&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;id&quot;)) %&gt;% separate(id, sep=&quot;_&quot;, into=c(&quot;gene&quot;, &quot;loc&quot;, &quot;strand&quot;, &quot;peak&quot;))</code></pre>
<pre><code>Warning: Expected 4 pieces. Additional pieces discarded in 4 rows [14735,
14736, 14737, 14738].</code></pre>
<pre class="r"><code>ind=colnames(nuclearPeakUs)[8:dim(nuclearPeakUs)[2]]
nuclearPeakUs_CountNum=read.table(&quot;../data/phenotype/APApeak_Phenotype_GeneLocAnno.Nuclear.CountsOnlyNumeric&quot;, col.names = ind)


#numeric with anno
nuclearPeak=as.data.frame(cbind(nuclearPeakUs[,1:7], nuclearPeakUs_CountNum))



nuclearPeak_inNew=nuclearPeak %&gt;% filter(peak %in% uniqnewPasnum$peak)

nuclearPeak_inNew_melt=melt(nuclearPeak_inNew,id.vars=c(&quot;chr&quot;, &quot;start&quot;,&quot;end&quot;, &quot;gene&quot;,&quot;loc&quot;, &quot;peak&quot;,&quot;strand&quot;),value.name = &quot;Usage&quot; ,variable.name = &quot;Ind&quot;) %&gt;% mutate(New15=ifelse(Ind %in% batch4$line, &quot;Yes&quot;, &quot;No&quot;))</code></pre>
<pre class="r"><code>nuclearPeak_inNew_meltGroup=nuclearPeak_inNew_melt %&gt;% group_by(New15,peak) %&gt;% summarise(meanUsage=mean(Usage))

ggplot(nuclearPeak_inNew_meltGroup, aes(x=New15, y=meanUsage, fill=New15))+geom_boxplot(width=.5) + theme(legend.position = &quot;none&quot;) + labs(x=&quot;Individuals in New batch&quot;, y=&quot;Mean Usage in Group&quot;, title=&quot;Nuclear Mean usage for new PAS in 39 ind v new 15&quot;)</code></pre>
<p><img src="figure/NewvOldPeakQC.Rmd/unnamed-chunk-34-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(nuclearPeak_inNew_meltGroup, aes(x=meanUsage, group=New15, fill=New15))+geom_density(alpha=.3)  + labs( x=&quot;Mean Usage in Group&quot;, title=&quot;Nuclear Mean usage for new PAS in 39 ind v new 15&quot;)+ scale_fill_discrete(name = &quot;Ind in new 15&quot;)</code></pre>
<p><img src="figure/NewvOldPeakQC.Rmd/unnamed-chunk-35-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="seperate-the-usage-for-old-data-by-ind-set" class="section level2">
<h2>Seperate the usage for old data by ind set</h2>
<pre class="r"><code>oldtotusage_annoM=melt(oldtotusage_anno, id.vars=c(&quot;chr&quot;,&quot;start&quot;,&quot;end&quot;,&quot;peakID&quot;), value.name = &quot;Usage&quot;, variable.name = &quot;Ind&quot;) %&gt;% mutate(New15=ifelse(Ind %in% batch4$line, &quot;Yes&quot;, &quot;No&quot;))

oldtotusage_annoM_new=oldtotusage_annoM %&gt;% semi_join(TotalPeakUSMeanClass_newonly, by=c(&quot;chr&quot;,&quot;start&quot;, &quot;end&quot;)) %&gt;%  group_by(New15,peakID) %&gt;% summarise(meanUsage=mean(Usage))


ggplot(oldtotusage_annoM_new, aes(x=New15, y=meanUsage, fill=New15))+geom_boxplot(width=.5) + theme(legend.position = &quot;none&quot;) + labs(x=&quot;Individuals in New batch&quot;, y=&quot;Mean Usage in Group&quot;, title=&quot;Total mean usage for new PAS in old data&quot;)</code></pre>
<p><img src="figure/NewvOldPeakQC.Rmd/unnamed-chunk-36-1.png" width="672" style="display: block; margin: auto;" /> This shows the peaks had super low usage in both sets of individuals.</p>
<p>spread</p>
<pre class="r"><code>oldtotusage_annoM_new_spread=oldtotusage_annoM_new %&gt;% spread(New15,meanUsage)


ggplot(oldtotusage_annoM_new_spread, aes(x=No, y=Yes))+ geom_point() +geom_density_2d() + geom_smooth(method = &quot;lm&quot;) + annotate(&quot;text&quot;, label=&quot;New=.7Old+.009&quot; ,x=.25,y=.75) + labs(x=&quot;Original 39 Mean&quot;, y=&quot;Original 15 Mean&quot;, title=&quot;Mean Usage in old data by Ind set&quot;)</code></pre>
<p><img src="figure/NewvOldPeakQC.Rmd/unnamed-chunk-37-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>summary(lm(data=oldtotusage_annoM_new_spread, Yes~No))</code></pre>
<pre><code>
Call:
lm(formula = Yes ~ No, data = oldtotusage_annoM_new_spread)

Residuals:
     Min       1Q   Median       3Q      Max 
-0.30595 -0.01113 -0.00586  0.00590  0.32477 

Coefficients:
            Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept) 0.009067   0.000217   41.78   &lt;2e-16 ***
No          0.727159   0.004500  161.58   &lt;2e-16 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 0.02478 on 14640 degrees of freedom
Multiple R-squared:  0.6407,    Adjusted R-squared:  0.6407 
F-statistic: 2.611e+04 on 1 and 14640 DF,  p-value: &lt; 2.2e-16</code></pre>
<p>Yes=.7(No)</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.5.1 (2018-07-02)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Scientific Linux 7.4 (Nitrogen)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] workflowr_1.3.0 forcats_0.3.0   stringr_1.3.1   dplyr_0.8.0.1  
 [5] purrr_0.3.2     readr_1.3.1     tidyr_0.8.3     tibble_2.1.1   
 [9] ggplot2_3.1.0   tidyverse_1.2.1 reshape2_1.4.3 

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.0       cellranger_1.1.0 compiler_3.5.1   pillar_1.3.1    
 [5] git2r_0.23.0     plyr_1.8.4       tools_3.5.1      digest_0.6.18   
 [9] lubridate_1.7.4  jsonlite_1.6     evaluate_0.12    nlme_3.1-137    
[13] gtable_0.2.0     lattice_0.20-38  pkgconfig_2.0.2  rlang_0.3.1     
[17] cli_1.0.1        rstudioapi_0.10  yaml_2.2.0       haven_1.1.2     
[21] withr_2.1.2      xml2_1.2.0       httr_1.3.1       knitr_1.20      
[25] hms_0.4.2        generics_0.0.2   fs_1.2.6         rprojroot_1.3-2 
[29] grid_3.5.1       tidyselect_0.2.5 glue_1.3.0       R6_2.3.0        
[33] readxl_1.1.0     rmarkdown_1.10   modelr_0.1.2     magrittr_1.5    
[37] whisker_0.3-2    MASS_7.3-51.1    backports_1.1.2  scales_1.0.0    
[41] htmltools_0.3.6  rvest_0.3.2      assertthat_0.2.0 colorspace_1.3-2
[45] labeling_0.3     stringi_1.2.4    lazyeval_0.2.1   munsell_0.5.0   
[49] broom_0.5.1      crayon_1.3.4    </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
