---
title: "Home"
site: workflowr::wflow_site
output:
  workflowr::wflow_html:
    toc: false
---

Welcome to my research website.


##Preprocessing:

In the preprocessing steps of the analysis I will go from the fastq files that come off the sequencer to the PAS I will use for the rest of the analysis. 

* [Process fastq to cleaned bam files](fastq2bam.html)

* [Process bam to PAS](bam2pas.html) 


##QC and base visualization:

These files create some of the QC metrics I used for the analysis.

* [Look at PAS usage differences by gene coverage](PASusageQC.html) 

* [Look at read distribution accross features](Readdistagainstfeatures.html) 

* [Make descriptive plots for called PAS](PASdescriptiveplots.html)  

* [Plot signal site dristributions](signalsiteanalysis.html)

* [Count and usage correlation](corrbetweenind.html)

* [Sequencing meta data by batch](seqmetadatanalysis.html)  

* [Understand new PAS](NewvOldPeakQC.html)  

##apaQTL
* [Call apaQTLs in both fractions](mapapaQTL.html)

* [Check for batch effects in new QTLs](newQTLheatmap.html)
