---
title: "Ratio of Intronic reads up and downstream of PAS"
author: "Briana Mittleman"
date: "6/7/2019"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

I am interested in finding examples of intronic PAS that show RNAseq signatures upstream of the PAS but not downstream. To do this I will create a ratio of reads upstream/reads downstream standardized by the length of the region (up/downstream). 

To do this I can use the work I did [previously](nucintronicanalysis.html). Here I assigned each intronic PAS to an intron. I will do this analysis with the total fraction because I will be looking at steady state RNA seq. 


```{r}
library(workflowr)
library(tidyverse)
```

##Total 
```{bash,eval=F}
mkdir ../data/intronRNAratio
```



```{r}

totIntronicPeaks=read.table("../data/peaks_5perc/APApeak_Peaks_GeneLocAnno.Total.5perc.fc", stringsAsFactors = F, header = F,col.names = c("chr", "start", "end", "gene", "loc", "strand", "peak", "avgUsage")) %>% filter(loc=="intron") 

pas2intronTot=read.table("../data/intron_analysis/TotalIntronPeaksontoIntrons.bed",col.names = c("intronCHR", "intronStart", "intronEnd", "gene", "score", "strand", "peakCHR", "peakStart", "peakEnd", "PeakID", "meanUsage", "peakStrand"))  %>% mutate(PASloc=ifelse(strand=="+", peakEnd, peakStart)) %>% dplyr::select(intronCHR,intronStart, intronEnd, gene, strand, PeakID, PASloc ,meanUsage)

write.table(pas2intronTot, "../data/intronRNAratio/TotalIntronicPAS2Intron.txt", quote = F, row.names = F, col.names = F, sep="\t")
```

Make upstream and downstream PAS saf files using python. 

```{bash,eval=F}
 python getIntronUpstreamPAS.py
 python getIntronDownstreamPAS.py
```


These make Bed and SAF files. I will use the SAF files for feature counts with all of the RNA seq.  These files are in /project2/yangili1/LCL/RNAseqGeuvadisBams/*.final.bam

```{bash,eval=F}
sbatch FC_intornUpandDownsteamPAS.sh
```

Downstream Results:

```{r}
downstream=read.table("../data/intronRNAratio/DownstreamIntron.fc", header = T,stringsAsFactors = F)
downstreamMean=rowSums(downstream[,7:ncol(downstream)])

downstreanMeanDF=as.data.frame(cbind(downstream[,1:6], downstreamMean)) %>% mutate(DownstreamMean_st=downstreamMean/Length) %>% select(Geneid,DownstreamMean_st )
```

Upstream Results: 

```{r}
upstream=read.table("../data/intronRNAratio/UpstreamIntron.fc", header = T,stringsAsFactors = F)
upstreamMean=rowSums(upstream[,7:ncol(upstream)])
upstreamMeanDF=as.data.frame(cbind(upstream[,1:6], upstreamMean)) %>% mutate(UpstreamMean_st=upstreamMean/Length) %>% select(Geneid,UpstreamMean_st )
```


Join Results:

I will use upstream - downstream instead of a ratio to deal with 0s
```{r}
UpandDown=upstreamMeanDF %>% inner_join(downstreanMeanDF, by="Geneid") %>% mutate(ratio=UpstreamMean_st-DownstreamMean_st) %>% arrange(desc(ratio)) %>% separate(Geneid, sep=":", into=c("PAS", "gene", "loc", "PASloc", "Usage"))

summary(UpandDown$ratio)
```


I want to know how many are positive:

```{r}
MoreUp=UpandDown %>% filter(ratio>0) 
summary(MoreUp$ratio)
nrow(MoreUp)
```
13903 examples where there are more reads upstream in the intron than downstream.  

To look for examples:

```{r}
head(MoreUp)
```

##Nuclear  

We also have nacent RNA seq. I will do this with nuclear intronic PAS. 


```{r}
nucIntronicPeaks=read.table("../data/peaks_5perc/APApeak_Peaks_GeneLocAnno.Nuclear.5perc.fc", stringsAsFactors = F, header = F,col.names = c("chr", "start", "end", "gene", "loc", "strand", "peak", "avgUsage")) %>% filter(loc=="intron")


pas2intronNuc=read.table("../data/intron_analysis/IntronPeaksontoIntrons.bed",col.names = c("intronCHR", "intronStart", "intronEnd", "gene", "score", "strand", "peakCHR", "peakStart", "peakEnd", "PeakID", "meanUsage", "peakStrand"))  %>% mutate(PASloc=ifelse(strand=="+", peakEnd, peakStart)) %>% dplyr::select(intronCHR,intronStart, intronEnd, gene, strand, PeakID, PASloc ,meanUsage)

write.table(pas2intronTot, "../data/intronRNAratio/NuclearIntronicPAS2Intron.txt", quote = F, row.names = F, col.names = F, sep="\t")
```


Make upstream and downstream PAS saf files using python. 

```{bash,eval=F}
python getUpstreamIntronNuclear.py
python getDownstreamIntronNuclear.py
```

These make Bed and SAF files. I will use the SAF files for feature counts with all of the nacent RNA seq. 

```{bash,eval=F}
sbatch FC_NucintornUpandDown.sh
```



Downstream Results:

```{r}
downstreamNuc=read.table("../data/intronRNAratio/NuclearDownstreamIntron.fc", col.names = c("Geneid","Chr","Start", "End", "Strand", "Length", "Downstream"), header = T,stringsAsFactors = F)%>% mutate(Downstream_st=Downstream/Length) %>% select(Geneid,Downstream_st )

upstreamNuc=read.table("../data/intronRNAratio/NuclearUpstreamIntron.fc", col.names = c("Geneid","Chr","Start", "End", "Strand", "Length", "Upstream"), header = T,stringsAsFactors = F)%>% mutate(Upstream_st=Upstream/Length) %>% select(Geneid,Upstream_st )


UpandDown_nuc=upstreamNuc %>% inner_join(downstreamNuc, by="Geneid") %>% mutate(ratio=Upstream_st-Downstream_st) %>% arrange(desc(ratio)) %>% separate(Geneid, sep=":", into=c("PAS", "gene", "loc", "PASloc", "Usage"))

summary(UpandDown_nuc$ratio)
```


```{r}
MoreUpNuc=UpandDown_nuc %>% filter(ratio>0) 
summary(MoreUpNuc$ratio)
nrow(MoreUpNuc)
```

```{r}
head(MoreUpNuc)
```

