---
title: "Explore Nascent RNA seq"
author: "Briana Mittleman"
date: "5/28/2019"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


```{r}
library(workflowr)
library(tidyverse)
library(reshape2)
```

We recently recieved nascent RNA seq from the Staley and Li lab collaboration. I want to explore this data in relation to my PAS.  

Data: /project2/yangili1/yangili/RNAseqmap/nascentRNAseq/1*.sorted.bam

```{r,eval=F}
mkdir ../data/NascentRNA
```


First I will merge all of the individuals. 

```{bash,eval=F}
sbatch mergeBamNascent.sh
sbatch bam2bw.sh ../data/NascentRNA/NascentRNAMerged.sort.bam ../data/NascentRNA/NascentRNAMerged.sort.bw
```


First just look at the data along gene bodies:  

(chromosomes have chr)

```{bash,eval=F}
sbatch NascentRNAdtPlot.sh
```


I want to look at these at my PAS. First I will need to add CHR to my pas in /project2/gilad/briana/apaQTL/data/PAS/

```{bash,eval=F}
sed -e 's/^/chr/' ../data/PAS/APAPAS_GeneLocAnno.5perc.bed > ../data/PAS/APAPAS_GeneLocAnno.5perc_withCHR.bed
```


```{bash,eval=F}
sbatch NascnetRNAdtPlotPAS.sh
```

Look at the PAS used more in total and nuclear:  

```{bash,eval=F}
sbatch NascentRNAdtPlotTotPAS.sh
sbatch NascentRNAdtPlotNucPAS.sh
```


Intronic PAS 

```{r}
pasIntron=read.table("../data/PAS/APAPAS_GeneLocAnno.5perc_withCHR.bed",col.names = c("Chr", "start", "end", "PeakID", "score", "strand")) %>% separate(PeakID, into=c("peaknum", "geneID"), sep=":") %>% separate(geneID, into=c("gene", "loc"), sep="_") %>% filter(loc=="intron") %>%  mutate(peakIDuse=paste(peaknum,gene, sep=":")) %>% select(Chr, start,end, peaknum, score, strand)

write.table(pasIntron, "../data/PAS/APAPAS_GeneLocAnno.5perc_withCHR_Intronic.bed", col.names = F, row.names = F, quote = F, sep="\t")

pasUTR=read.table("../data/PAS/APAPAS_GeneLocAnno.5perc_withCHR.bed",col.names = c("Chr", "start", "end", "PeakID", "score", "strand")) %>% separate(PeakID, into=c("peaknum", "geneID"), sep=":") %>% separate(geneID, into=c("gene", "loc"), sep="_") %>% filter(loc=="utr3")  %>%  mutate(peakIDuse=paste(peaknum,gene, sep=":"))%>% select(Chr, start,end, peakIDuse, score, strand)

write.table(pasUTR, "../data/PAS/APAPAS_GeneLocAnno.5perc_withCHR_3UTR.bed", col.names = F, row.names = F, quote = F, sep="\t")
```



```{bash,eval=F}
sbatch NascentRNAdtPlot3UTRPAS.sh
sbatch NascentRNAdtPlotintronicPAS.sh
```


For the intronic PAS I want to seperate by those in the first intron and not.  

```{r}
allPAS=read.table("../data/PAS/APAPAS_GeneLocAnno.5perc_withCHR.bed", col.names = c("chr", "start", "end", "peak", "score", "strand")) %>% separate(peak, into=c("peakNumber", "geneloc"), sep=":") %>% mutate(PeakNum=paste("peak", peakNumber, sep=""))
peakIntron=read.table(file="../data/intron_analysis/PeakIdwithIntronID.txt", header = T) %>% separate(PeakID, into=c("PeakNum", "gene", "loc"),sep=":") %>% select(PeakNum, Intornid)
```

Join these to write seperate bed files:

```{r}
allPASwIntron=allPAS %>% inner_join(peakIntron, by="PeakNum")

FirstIntron= allPASwIntron %>% filter(Intornid==1 ) %>% select(-PeakNum,-Intornid)
write.table(FirstIntron, "../data/PAS/APAPAS_GeneLocAnno.5perc_withCHR_FirstIntron.bed",col.names = F, row.names = F, quote = F, sep="\t")

NotfirstIntron= allPASwIntron %>% filter(Intornid!=1 ) %>% select(-PeakNum,-Intornid)
write.table(NotfirstIntron, "../data/PAS/APAPAS_GeneLocAnno.5perc_withCHR_ExcludeFirstIntron.bed",col.names = F, row.names = F, quote = F, sep="\t")
```

```{bash,eval=F}
sbatch NascentRNAdtPlotFirstintronicPAS.sh
sbatch NascentRNAdtPlotExcludeFirstintronicPAS.sh

```


I also want to quantify nascent seq at each gene for comparing to other data types. I will use feature counts.  

I need to add the chr to /project2/gilad/briana/genome_anotation_data/refseq.ProteinCoding.noCHR.SAF

```{r}
genes=read.table("/project2/gilad/briana/genome_anotation_data/refseq.ProteinCoding.noCHR.SAF", header = T, stringsAsFactors = F) %>% mutate(Chr=paste("chr", Chr, sep=""))
write.table(genes,"/project2/gilad/briana/genome_anotation_data/refseq.ProteinCoding.SAF", col.names = T, row.names = F, quote = F, sep="\t")

```

```{bash,eval=F}
sbatch FC_nascentseq.sh
```


