---
title: "New QTL heatmap"
author: "Briana Mittleman"
date: "4/29/2019"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


```{r}
library(reshape2)
library(gdata)
library(workflowr)
library(gplots)
library(tidyverse)
library(cowplot)
```

##Compare QTLs to those found with previous batch data  

I have about double the QTLs hear compared to before resequencing batch 4. I will look at the new QTL to see if there is evidence for them being false positives. I am going to see if there is structure in the genotypes for these QTLs.  

The old QTLs are from the threeprimeseq repository.  

###Total

Import old QTLs
```{r}
oldtot=read.table("../../threeprimeseq/data/perm_APAqtl_GeneLocAnno_noMP_5percUs/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno.NoMP_sm_quant.Total.fixed.pheno_5perc_permResBH.txt", header=T,stringsAsFactors = F) %>% separate(pid, into=c("Chr", "Start", "End", "PeakID"), sep=":") %>% separate(PeakID, into=c("Gene", "Strand","Peak"), sep="_")

OldTotQTLs= oldtot %>% filter(-log10(bh)>=1)
nrow(OldTotQTLs)
```
Import new QTLs:  

```{r}
newTotQTLs=read.table("../data/apaQTLs/Total_apaQTLs_5fdr.txt", stringsAsFactors = F, header = T)
nrow(newTotQTLs)
```

Filter out those matching from the old:  

```{r}
UniqueNewTot=newTotQTLs %>% semi_join(OldTotQTLs, by="sid")
```

There are only 105 new snps This makes sense because 1 sno associates with multiple peaks. 

Write these out to fetch the genotypes:  

```{r}
write.table(UniqueNewTot, file="../data/apaQTLs/Total_apaQTLs_5fdr_NewUnique.txt", quote = F, col.names = F, row.names = F)
```

###Nuclear

```{r}
oldnuc=read.table("../../threeprimeseq/data/perm_APAqtl_GeneLocAnno_noMP_5percUs/filtered_APApeaks_merged_allchrom_refseqGenes.GeneLocAnno.NoMP_sm_quant.Nuclear.fixed.pheno_5perc_permResBH.txt", header=T,stringsAsFactors = F) %>% separate(pid, into=c("Chr", "Start", "End", "PeakID"), sep=":") %>% separate(PeakID, into=c("Gene", "Strand","Peak"), sep="_")

OldNucQTLs= oldnuc %>% filter(-log10(bh)>=1)
nrow(OldNucQTLs)
```
Import new QTLs:  

```{r}
newNucQTLs=read.table("../data/apaQTLs/Nuclear_apaQTLs_5fdr.txt", stringsAsFactors = F, header = T)
nrow(newNucQTLs)
```

Filter out those matching from the old:  

```{r}
UniqueNewNuc=newNucQTLs %>% semi_join(OldNucQTLs, by="sid")
```


There are 200 new snps in this set.  
```{r}
write.table(UniqueNewNuc, file="../data/apaQTLs/Nuclear_apaQTLs_5fdr_NewUnique.txt", quote = F, col.names = F, row.names = F)
```



##Extract genotypes: 

I wrote a script to pull the doses from the vcf file. Run it with:

```{bash,eval=F}
 python extractGenotypes.py ../data/apaQTLs/Nuclear_apaQTLs_5fdr_NewUnique.txt ../data/QTLGenotypes/Genotypes_NuclearapaQTLS_newunique.txt
 
  python extractGenotypes.py ../data/apaQTLs/Total_apaQTLs_5fdr_NewUnique.txt ../data/QTLGenotypes/Genotypes_TotalapaQTLS_newunique.txt
  
```

I also need the header from the VCF to have the individuals:  

```{bash,eval=F}
head -n14 /project2/gilad/briana/YRI_geno_hg19/allChrom.dose.filt.vcf | tail -n1  > ../data/QTLGenotypes/vcfheader.txt

#manually remove # and unneaded columns, keep snp and ind. 
```

```{r}
vcfhead=read.table("../data/QTLGenotypes/vcfheader.txt", header = T)
```

input sample list:  

```{r}
samples=read.table("../data/phenotype/SAMPLE.txt")
samplist=as.vector(samples$V1)
```


###Total: 

```{r}
totgeno=read.table("../data/QTLGenotypes/Genotypes_TotalapaQTLS_newunique.txt", col.names = colnames(vcfhead)) %>% select(samplist) %>% t()
```

Correlation:  

```{r}
totgeneCorr=round(cor(totgeno),2)

heatmap.2(as.matrix(totgeneCorr),trace="none", dendrogram =c("none"), main="Genotype correlation\n for new Total QTL snps")
```
###Nuclear  

```{r}
nucgeno=read.table("../data/QTLGenotypes/Genotypes_NuclearapaQTLS_newunique.txt", col.names = colnames(vcfhead)) %>% select(samplist) %>% t()
```

Correlation:  

```{r}
nucgeneCorr=round(cor(nucgeno),2)

heatmap.2(as.matrix(nucgeneCorr),trace="none", dendrogram =c("none"),main="Genotype correlation \n for new Nuclear QTL snps")
```

##Structure in all QTLs
This was for the new QTLs. I want to see if there is structure more generally.  

```{bash,eval=F}
 python extractGenotypes.py ../data/apaQTLs/Nuclear_apaQTLs_5fdr.txt ../data/QTLGenotypes/Genotypes_NuclearapaQTLS_ALLQTLs.txt

 python extractGenotypes.py ../data/apaQTLs/Total_apaQTLs_5fdr.txt ../data/QTLGenotypes/Genotypes_TotalapaQTLS_ALLQTLs.txt
```


```{r}
totgenoall=read.table("../data/QTLGenotypes/Genotypes_TotalapaQTLS_ALLQTLs.txt", col.names = colnames(vcfhead)) %>% select(samplist) %>% t()

totgenoallsm=totgenoall[,1:300]

totgeneallCorr=round(cor(totgenoallsm),2)

heatmap.2(as.matrix(totgeneallCorr),trace="none", dendrogram =c("none"),main="Genotype correlation \n for first 300 Total QTL snps")

```

```{r}
nucgenoall=read.table("../data/QTLGenotypes/Genotypes_NuclearapaQTLS_ALLQTLs.txt", col.names = colnames(vcfhead)) %>% select(samplist) %>% t()
nucgenoallsm=nucgenoall[,1:400]


nucgeneallCorr=round(cor(nucgenoallsm),2)

heatmap.2(as.matrix(nucgeneallCorr),trace="none", dendrogram =c("none"),main="Genotype correlation \n for first 400 Nuclear QTL snps")


```


##Compare beta values in 55 vs 39  

I want to make a scatter plot comaparring the new QTL associations in the 55 vs 39 individauls. If the qtls are real we expect a high correlation.  

To do this I can recall the qtls with a smaller sample list excluding the 15 new individuals. 

I need to make a list of the individuals not in the 4th batch.

```{r}
batch1.2.3=read.table("../data/MetaDataSequencing.txt", header=T,stringsAsFactors = F)%>% filter(fraction=="total") %>%  select(line, batch) %>% filter(batch != 4)
```


```{r}
samplelist=read.table("../data/phenotype/SAMPLE.txt", col.names = c("line"),stringsAsFactors = F)
```

Make a new directory for the 39ind qtls:  

```{bash,eval=F}
mkdir ../data/ThirtyNineIndQtl_nominal
```

Filter the sample list 

```{r}
samplelist_39= samplelist %>% semi_join(batch1.2.3, by="line")
```

```{r}
write.table(samplelist_39, file="../data/ThirtyNineIndQtl_nominal/samplelist39.txt", col.names = F, row.names = F, quote = F)
```

Run the QTL code with this sample list 
```{bash,eval=F}
sbatch aAPAqtl_nominal39ind.sh

```

Concatinate results:  

```{bash,eval=F}
cat APApeak_Phenotype_GeneLocAnno.Total.5perc.fc.gz.qqnorm_chr* > APApeak_Phenotype_GeneLocAnno.Total.5perc.fc.gz.qqnorm_AllChr.txt

cat APApeak_Phenotype_GeneLocAnno.Nuclear.5perc.fc.gz.qqnorm_chr* > APApeak_Phenotype_GeneLocAnno.Nuclear.5perc.fc.gz.qqnorm_AllChr.txt

```



I want to filter the results for the new snps in the uniquenewtot. These results are in  data/apaQTLs


I need to write a script that makes a dictionary with each of the new QTLs in the format above. Then I can run throguh the nominal values and keep only the values in the dictionary.  

I can run this on the 55 and 39 nominal files then combine the files to create the scatterplot.  


###total
```{bash,eval=F}
python selectNominalPvalues.py ../data/apaQTLs/Total_apaQTLs_5fdr_NewUnique.txt ../data/ThirtyNineIndQtl_nominal/APApeak_Phenotype_GeneLocAnno.Total.5perc.fc.gz.qqnorm_AllChr.txt ../data/ThirtyNineIndQtl_nominal/Total_apaQTLs_NewUniqNom_37ing.txt

python selectNominalPvalues.py ../data/apaQTLs/Total_apaQTLs_5fdr_NewUnique.txt ../data/apaQTLNominal/APApeak_Phenotype_GeneLocAnno.Total.5perc.fc.gz.qqnorm_AllChr.txt ../data/ThirtyNineIndQtl_nominal/Total_apaQTLs_NewUniqNom_55ind.txt

```
Import files:

```{r}
newin37_tot=read.table("../data/ThirtyNineIndQtl_nominal/Total_apaQTLs_NewUniqNom_37ing.txt",col.names=c("peakID", "snp", "dist", "Nompval39","Beta39"), stringsAsFactors = F)

newin54_tot=read.table("../data/ThirtyNineIndQtl_nominal/Total_apaQTLs_NewUniqNom_55ind.txt",col.names=c("peakID", "snp", "dist", "Nompval54","Beta54"), stringsAsFactors = F) 
```
Join these:
```{r}
newinboth=newin54_tot %>% full_join(newin37_tot, by=c("peakID", "snp"))
```

```{r}
total_qtlind=ggplot(newinboth,aes(x=Beta54, y=Beta39)) + geom_point()  + labs(title="New Total apaQTLs \nin different ind. sets", ylab="Beta 39 ind", xlab="Beta 55 ind")

```

###nuclear  

```{bash,eval=F}
python selectNominalPvalues.py ../data/apaQTLs/Nuclear_apaQTLs_5fdr_NewUnique.txt ../data/ThirtyNineIndQtl_nominal/APApeak_Phenotype_GeneLocAnno.Nuclear.5perc.fc.gz.qqnorm_AllChr.txt ../data/ThirtyNineIndQtl_nominal/Nuclear_apaQTLs_NewUniqNom_37ing.txt

python selectNominalPvalues.py ../data/apaQTLs/Nuclear_apaQTLs_5fdr_NewUnique.txt ../data/apaQTLNominal/APApeak_Phenotype_GeneLocAnno.Nuclear.5perc.fc.gz.qqnorm_AllChr.txt ../data/ThirtyNineIndQtl_nominal/Nuclear_apaQTLs_NewUniqNom_55ind.txt
```

```{r}
newin37_nuc=read.table("../data/ThirtyNineIndQtl_nominal/Nuclear_apaQTLs_NewUniqNom_37ing.txt",col.names=c("peakID", "snp", "dist", "Nompval39","Beta39"), stringsAsFactors = F)

newin54_nuc=read.table("../data/ThirtyNineIndQtl_nominal/Nuclear_apaQTLs_NewUniqNom_55ind.txt",col.names=c("peakID", "snp", "dist", "Nompval54","Beta54"), stringsAsFactors = F)
```
Join these:
```{r}
newinboth_nuc=newin54_nuc %>% full_join(newin37_nuc, by=c("peakID", "snp"))
```

```{r}
nuclear_qtlind=ggplot(newinboth_nuc,aes(x=Beta54, y=Beta39)) + geom_point()  + labs(title="New Nuclear apaQTLs\n in different ind. sets", ylab="Beta 39 ind", xlab="Beta 55 ind")

```


plot both: 

```{r}
plot_grid(total_qtlind, nuclear_qtlind)
```

Plot -log10pvalue  

```{r}
pvalplot_tot=ggplot(newinboth,aes(x=-log10(Nompval54), y=-log10(Nompval39))) + geom_point()  + labs(title="New Total apaQTLs \nin different ind. sets", y="-log10 pval 39 ind", x="-log10 pval 55 ind")
```
```{r}
pvalplot_nuc=ggplot(newinboth_nuc,aes(x=-log10(Nompval54), y=-log10(Nompval39))) + geom_point()  + labs(title="New Nuclear apaQTLs \nin different ind. sets", y="-log10 pval 39 ind", x="-log10 pval 55 ind")

```

```{r}
plot_grid(pvalplot_tot,pvalplot_nuc)
```
Confirm these QTLs are in new phenotype peaks:  

```{r}
newvold=read.table("../data/peaks_5perc/NewVOldPeaks.txt", header = T, stringsAsFactors = F) %>% select(peak, New)

colnames(newvold)=c("Peak", "Set")
```
Are the peaks in UniqueNewTot and UniqueNewNuc 

```{r}
UniqueNewTot_set= UniqueNewTot %>% inner_join(newvold, by="Peak")


UniqueNewTot_set$Set=as.factor(UniqueNewTot_set$Set)
summary(UniqueNewTot_set$Set)
```

```{r}
UniqueNewNuc_set= UniqueNewNuc %>% inner_join(newvold, by="Peak")


UniqueNewNuc_set$Set=as.factor(UniqueNewNuc_set$Set)
summary(UniqueNewNuc_set$Set)
```
##Example boxplots  

I want to plot these  qtls as boxplots. I will plot usage vs. genotype. I want to color the new 15 individuals differently.  


DCLRE1C peak14470 10:14974843 
```{bash,eval=F}
#get genotype from vcf
less chr10.dose.filt.vcf.gz  |  grep 10:14974843 > ../apaQTL/data/exampleQTLs/Geno_10:14974843.txt

#get pheno from pheno_5perc
less APApeak_Phenotype_GeneLocAnno.Nuclear.5perc.fc.gz | grep peak14470 > ../exampleQTLs/Pheno_peak14470.txt
```

 col.names = colnames(vcfhead))

```{bash,eval=F}

head -n14 /project2/gilad/briana/YRI_geno_hg19/allChrom.dose.filt.vcf | tail -n1  > ../data/exampleQTLs/vcfheader.txt

#get rid of # manually 

#pheno head

 less APApeak_Phenotype_GeneLocAnno.Total.5perc.fc.gz | head -n1 > ../exampleQTLs/phenoHead.txt
```



```{r}
vcfheadfull=read.table("../data/exampleQTLs/vcfheader.txt", header = T)
phenohead=read.table("../data/exampleQTLs/phenoHead.txt", header = T)

```

Try first

```{r}
geno_peak14470=read.table("../data/exampleQTLs/Geno_10:14974843.txt", col.names = colnames(vcfheadfull), stringsAsFactors = F) %>% select(-CHROM, -POS, -REF, -ALT, -QUAL, -FILTER, -INFO, -FORMAT)
geno_peak14470M=melt(geno_peak14470, id.vars = "ID") %>% separate(value, into=c("genotype", "extra", "extra2"), sep=":") %>% mutate(dose=round(as.integer(extra), digits = 1)) %>% select(-genotype, -extra, -extra2)

geno_peak14470M$variable=as.character(geno_peak14470M$variable)

pheno_peak14470=read.table("../data/exampleQTLs/Pheno_peak14470.txt", col.names = colnames(phenohead), stringsAsFactors = F) 
pheno_peak14470M=melt(pheno_peak14470,id.vars = "chrom") %>% separate(value, into=c("num", "dom"), sep="/") %>% mutate(usage=as.integer(num)/as.integer(dom))

pheno_peak14470M$variable=as.character(pheno_peak14470M$variable)
```

Join these 


```{r}
genoPheno14470=pheno_peak14470M %>% inner_join(geno_peak14470M, by="variable") %>% mutate(set=ifelse(variable %in% samplelist_39$line, "old39", "new15"))

genoPheno14470$set=as.factor(genoPheno14470$set)
genoPheno14470$dose=as.factor(genoPheno14470$dose)
```


```{r}
ggplot(genoPheno14470, aes(x=dose, y=usage)) + geom_boxplot()+ geom_jitter(aes(col=set))
```
Try 1 more: 
peak33185 12:116544223

```{bash,eval=F}
#get genotype from vcf
less /project2/gilad/briana/YRI_geno_hg19/chr12.dose.filt.vcf.gz  |  grep 12:116544223 > ../data/exampleQTLs/Geno_12:116544223.txt

#get pheno from pheno_5perc
less ../data/phenotype_5perc/APApeak_Phenotype_GeneLocAnno.Nuclear.5perc.fc.gz | grep peak33185 > ../data/exampleQTLs/Pheno_peak33185.txt
```


```{r}
geno_peak33185=read.table("../data/exampleQTLs/Geno_12:116544223.txt", col.names = colnames(vcfheadfull), stringsAsFactors = F) %>% select(-CHROM, -POS, -REF, -ALT, -QUAL, -FILTER, -INFO, -FORMAT)
geno_peak33185M=melt(geno_peak33185, id.vars = "ID") %>% separate(value, into=c("genotype", "extra", "extra2"), sep=":") %>% mutate(dose=round(as.integer(extra), digits = 1)) %>% select(-genotype, -extra, -extra2)

geno_peak33185M$variable=as.character(geno_peak33185M$variable)

pheno_peak33185=read.table("../data/exampleQTLs/Pheno_peak33185.txt", col.names = colnames(phenohead), stringsAsFactors = F) 
pheno_peak33185M=melt(pheno_peak33185,id.vars = "chrom") %>% separate(value, into=c("num", "dom"), sep="/") %>% mutate(usage=as.integer(num)/as.integer(dom))

geno_peak33185M$variable=as.character(geno_peak33185M$variable)
```

Join these 


```{r}
genoPhenopeak33185=pheno_peak33185M %>% inner_join(geno_peak33185M, by="variable") %>% mutate(set=ifelse(variable %in% samplelist_39$line, "old39", "new15"))

genoPhenopeak33185$set=as.factor(genoPhenopeak33185$set)
genoPhenopeak33185$dose=as.factor(genoPhenopeak33185$dose)
```


```{r}
ggplot(genoPhenopeak33185, aes(x=dose, y=usage)) + geom_boxplot()+ geom_jitter(aes(col=set))
```


peak59741 17:78371306


```{bash,eval=F}
#get genotype from vcf
less /project2/gilad/briana/YRI_geno_hg19/chr17.dose.filt.vcf.gz  |  grep 17:78371306 > ../data/exampleQTLs/Geno_17:78371306.txt

#get pheno from pheno_5perc
less ../data/phenotype_5perc/APApeak_Phenotype_GeneLocAnno.Nuclear.5perc.fc.gz | grep peak59741 > ../data/exampleQTLs/Pheno_peak59741.txt
```


```{r}
geno_peak59741=read.table("../data/exampleQTLs/Geno_17:78371306.txt", col.names = colnames(vcfheadfull), stringsAsFactors = F) %>% select(-CHROM, -POS, -REF, -ALT, -QUAL, -FILTER, -INFO, -FORMAT)
geno_peak59741M=melt(geno_peak59741, id.vars = "ID") %>% separate(value, into=c("genotype", "extra", "extra2"), sep=":") %>% mutate(dose=round(as.integer(extra), digits = 1)) %>% select(-genotype, -extra, -extra2)

geno_peak59741M$variable=as.character(geno_peak59741M$variable)

pheno_peak59741=read.table("../data/exampleQTLs/Pheno_peak59741.txt", col.names = colnames(phenohead), stringsAsFactors = F) 
pheno_peak59741M=melt(pheno_peak59741,id.vars = "chrom") %>% separate(value, into=c("num", "dom"), sep="/") %>% mutate(usage=as.integer(num)/as.integer(dom))

geno_peak59741M$variable=as.character(geno_peak59741M$variable)
```

Join these 


```{r}
genoPhenopeak59741=pheno_peak59741M %>% inner_join(geno_peak59741M, by="variable") %>% mutate(set=ifelse(variable %in% samplelist_39$line, "old39", "new15"))

genoPhenopeak59741$set=as.factor(genoPhenopeak59741$set)
genoPhenopeak59741$dose=as.factor(genoPhenopeak59741$dose)
```


```{r}
ggplot(genoPhenopeak59741, aes(x=dose, y=usage)) + geom_boxplot()+ geom_jitter(aes(col=set))
```
