---
title: "Usage Effect Sizes"
author: "Briana Mittleman"
date: "5/24/2019"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

In order to compare effect sizes for the QTLs I have previously identified in an interpretable manner, I need to run the linear model with the non normalized usage. To do this I will separate the the usage (with annotation) files by chromosome and run fastqtl on these files.  

```{r}
library(workflowr)
library(tidyverse)
library(cowplot)
```

##Prepare files  

countsnum= APApeak_Phenotype_GeneLocAnno.Total.5perc.CountsNumeric, APApeak_Phenotype_GeneLocAnno.Nuclear.5perc.CountsNumeric

id file= APApeak_Phenotype_GeneLocAnno.Nuclear.5perc.fc.gz, APApeak_Phenotype_GeneLocAnno.Total.5perc.fc.gz



```{r}
totAnno= read.table("../data/phenotype_5perc/APApeak_Phenotype_GeneLocAnno.Total.5perc.fc.gz", stringsAsFactors = F, header = T) %>% separate(chrom, into=c("Chrchrom", "Start", "End", "ID"),sep=":") %>% mutate(Chr=str_sub(Chrchrom, 4, str_length(Chrchrom)))
                                                                                                                                                                                                                
colnamesTot= colnames(totAnno)[5:58]
totUsage=read.table("../data/phenotype_5perc/APApeak_Phenotype_GeneLocAnno.Total.5perc.CountsNumeric", stringsAsFactors = F, header = F, col.names = colnamesTot) 

totUsageAnno=as.data.frame(cbind(Chr=totAnno$Chr, start=totAnno$Start, end=totAnno$End, ID=totAnno$ID, totUsage ))

write.table(totUsageAnno,file="../data/nonNorm_pheno/TotalUsageAllChrom.txt", col.names = T, row.names = F, quote = F, sep="\t" )
```



```{r}
nucAnno= read.table("../data/phenotype_5perc/APApeak_Phenotype_GeneLocAnno.Nuclear.5perc.fc.gz", stringsAsFactors = F, header = T)%>% separate(chrom, into=c("Chrchrom", "Start", "End", "ID"),sep=":") %>% mutate(Chr=str_sub(Chrchrom, 4, str_length(Chrchrom)))
colnamesNuc= colnames(nucAnno)[5:58]
nucUsage=read.table("../data/phenotype_5perc/APApeak_Phenotype_GeneLocAnno.Nuclear.5perc.CountsNumeric", stringsAsFactors = F, header = F, col.names = colnamesNuc) 


nucUsageAnno=as.data.frame(cbind(Chr=nucAnno$Chr, start=nucAnno$Start, end=nucAnno$End, ID=nucAnno$ID, nucUsage ))

write.table(nucUsageAnno,file="../data/nonNorm_pheno/NuclearUsageAllChrom.txt", col.names = T, row.names = F, quote = F, sep="\t" )
```


##Run QTL scripts
I will create a python script to seperate the file into each chromosome for running fastQTL.  

```{bash,eval=F}
sbatch run_sepUsagephen.sh
sbatch ZipandTabPheno.sh
sbatch ApaQTL_nominalNonnorm.sh
```

Concatinate files:  

```{bash,eval=F}
cat  TotalUsageChrom*.nominal.out > TotalUsageChrom_Nominal_AllChrom.txt
cat  NuclearUsageChrom*.nominal.out > NuclearUsageChrom_Nominal_AllChrom.txt

```

##Pull out real total and nuc QLTs  


```{bash,eval=F}
python qtlsPvalOppFrac.py ../data/apaQTLs/Nuclear_apaQTLs4pc_5fdr.txt ../data/nonNorm_pheno/TotalUsageChrom_Nominal_AllChrom.txt ../data/QTLoverlap_nonNorm/NuclearQTLinTotalNominal_nonNorm.txt  

python qtlsPvalOppFrac.py ../data/apaQTLs/Nuclear_apaQTLs4pc_5fdr.txt ../data/nonNorm_pheno/NuclearUsageChrom_Nominal_AllChrom.txt ../data/QTLoverlap_nonNorm/NuclearQTLinNuclearNominal_nonNorm.txt  


python qtlsPvalOppFrac.py ../data/apaQTLs/Total_apaQTLs4pc_5fdr.txt ../data/nonNorm_pheno/TotalUsageChrom_Nominal_AllChrom.txt ../data/QTLoverlap_nonNorm/TotalQTLinTotalNominal_nonNorm.txt  

python qtlsPvalOppFrac.py ../data/apaQTLs/Total_apaQTLs4pc_5fdr.txt ../data/nonNorm_pheno/NuclearUsageChrom_Nominal_AllChrom.txt ../data/QTLoverlap_nonNorm/TotalQTLinNuclearNominal_nonNorm.txt  
```



```{r}
totAPAinNuc=read.table("../data/QTLoverlap_nonNorm/TotalQTLinNuclearNominal_nonNorm.txt", header = F, stringsAsFactors = F, col.names=c("peakID", "snp", "dist", "pval", "slope"))


nucAPAinTot=read.table("../data/QTLoverlap_nonNorm/NuclearQTLinTotalNominal_nonNorm.txt", header = F, stringsAsFactors = F, col.names=c("peakID", "snp", "dist", "pval", "slope"))

totAPAinTot=read.table("../data/QTLoverlap_nonNorm/TotalQTLinTotalNominal_nonNorm.txt", header = F, stringsAsFactors = F, col.names=c("peakID", "snp", "dist", "pval", "slope")) %>% dplyr::select(peakID, snp, slope) %>% dplyr::rename("Originalslope"=slope)

nucAPAinNuc=read.table("../data/QTLoverlap_nonNorm/NuclearQTLinNuclearNominal_nonNorm.txt", header = F, stringsAsFactors = F, col.names=c("peakID", "snp", "dist", "pval", "slope")) %>% dplyr::select(peakID, snp, slope)%>% dplyr::rename("Originalslope"=slope)
```



##Total

```{r}
TotBoth= totAPAinNuc %>% inner_join(totAPAinTot,by=c("peakID", "snp"))

summary(lm(TotBoth$slope ~ TotBoth$Originalslope))


totbothplot=ggplot(TotBoth, aes(x=Originalslope, y=slope))+geom_point() + geom_smooth(method="lm") + labs(title="Total apaQTL effect sizes", x="Effect size in Total",y="Effect size in Nucler") + geom_density_2d(col="red") + annotate("text", y=2, x=2, label="R2=.61, slope=0.4")
```
##Nuclear  

```{r}
NucBoth= nucAPAinTot %>% inner_join(nucAPAinNuc,by=c("peakID", "snp"))
summary(lm(NucBoth$slope ~ NucBoth$Originalslope))

Nucbothplot=ggplot(NucBoth, aes(x=Originalslope, y=slope))+geom_point() + geom_smooth(method="lm") + labs(title="Nuclear apaQTL effect sizes", x="Effect size in Nuclear",y="Effect size in Total") + geom_density_2d(col="red") +  annotate("text", y=2, x=1, label="R2=.81, slope=0.74")
```

```{r}
plot_grid(totbothplot,Nucbothplot)
```

##Remove Effect size > abs(1)  


```{r}
totAPAinNucFilt=read.table("../data/QTLoverlap_nonNorm/TotalQTLinNuclearNominal_nonNorm.txt", header = F, stringsAsFactors = F, col.names=c("peakID", "snp", "dist", "pval", "slope")) %>% filter(abs(slope)<= 1)


nucAPAinTotFilt=read.table("../data/QTLoverlap_nonNorm/NuclearQTLinTotalNominal_nonNorm.txt", header = F, stringsAsFactors = F, col.names=c("peakID", "snp", "dist", "pval", "slope")) %>% filter(abs(slope)<= 1)

totAPAinTotFilt=read.table("../data/QTLoverlap_nonNorm/TotalQTLinTotalNominal_nonNorm.txt", header = F, stringsAsFactors = F, col.names=c("peakID", "snp", "dist", "pval", "slope")) %>% dplyr::select(peakID, snp, slope) %>%   filter(abs(slope)<= 1) %>% dplyr::rename("Originalslope"=slope)

nucAPAinNucFilt=read.table("../data/QTLoverlap_nonNorm/NuclearQTLinNuclearNominal_nonNorm.txt", header = F, stringsAsFactors = F, col.names=c("peakID", "snp", "dist", "pval", "slope")) %>% dplyr::select(peakID, snp, slope)%>%  filter(abs(slope)<= 1) %>%dplyr::rename("Originalslope"=slope)
```

```{r}
TotBothFilt= totAPAinNucFilt %>% inner_join(totAPAinTotFilt,by=c("peakID", "snp"))

summary(lm(TotBothFilt$slope ~ TotBothFilt$Originalslope))


totbothplotfilt=ggplot(TotBothFilt, aes(x=Originalslope, y=slope))+geom_point() + geom_smooth(method="lm") + labs(title="Total apaQTL effect sizes", y="Effect size in Nuclear",x="Effect size in Total") + geom_density_2d(col="red") + annotate("text", y=.75, x=.1, label="R2=.76, slope=0.77")+ geom_abline(slope=1,color="green")
```



```{r}
NucBothFilt= nucAPAinTotFilt %>% inner_join(nucAPAinNucFilt,by=c("peakID", "snp"))
summary(lm(NucBothFilt$slope ~ NucBothFilt$Originalslope))

Nucbothplotfilt=ggplot(NucBothFilt, aes(x=Originalslope, y=slope))+geom_point() + geom_smooth(method="lm") + labs(title="Nuclear apaQTL effect sizes", y="Effect size in Total",x="Effect size in Nuclear") + geom_density_2d(col="red") +  annotate("text", y=.75, x=.1, label="R2=.73, slope=0.71")+ geom_abline(slope=1,color="green")
```



```{r}
plot_grid(totbothplotfilt,Nucbothplotfilt)
```


###Box plots to look at the outliers:  


get headers:

```{bash,eval=F}
less /project2/gilad/briana/YRI_geno_hg19/chr3.dose.filt.vcf.gz | head -n14 | tail -n1 > /project2/gilad/briana/apaQTL/data/ExampleQTLPlots/genotypeHeader.txt
less /project2/gilad/briana/apaQTL/data/phenotype_5perc/APApeak_Phenotype_GeneLocAnno.Total.5perc.fc.gz | head -n1 > /project2/gilad/briana/apaQTL/data/ExampleQTLPlots/phenotypeHeader.txt
```

remove the hashtag in these

apaqtlfacetboxplots.R
```{r,eval=F}
library(tidyverse)
library(reshape2)
library(optparse)
library(cowplot)


option_list = list(
  make_option(c("-P", "--pheno"), action="store", default=NA, type='character',
              help="input pheno file"),
  make_option(c("-G", "--geno"), action="store", default=NA, type='character',
              help="input genotype"),
  make_option(c("-g", "--gene"), action="store", default=NA, type='character',
              help="gene"),
  make_option(c("-p", "--peakID"), action="store", default=NA, type='character',
              help="peakID"),
  make_option(c("-o", "--output"), action="store", default=NA, type='character',
              help="output file for plot")
)

opt_parser <- OptionParser(option_list=option_list)
opt <- parse_args(opt_parser)


opt_parser <- OptionParser(option_list=option_list)
opt <- parse_args(opt_parser)


phenohead=read.table("/project2/gilad/briana/apaQTL/data/ExampleQTLPlots/phenotypeHeader.txt", header = T,stringsAsFactors = F)
pheno=read.table(opt$pheno, col.names =colnames(phenohead),stringsAsFactors = F)


meltpheno=melt(pheno, id.vars = "chrom", value.name = "Ratio", variable.name = "Individual") %>% separate(Ratio, into=c("num", "denom"), sep="/")  %>%   separate(chrom, into=c("chrom", "start", "end", "peakID"),sep=":") %>% mutate(PeakLoc=paste(start, end, sep=":"))

meltpheno$Individual= as.character(meltpheno$Individual)
meltpheno$num= as.numeric(meltpheno$num)
meltpheno$denom=as.numeric(meltpheno$denom)


genoHead=read.table("/project2/gilad/briana/apaQTL/data/ExampleQTLPlots/genotypeHeader.txt", header = T,stringsAsFactors = F)
geno=read.table(opt$geno, col.names =colnames(genoHead),stringsAsFactors = F ) %>% select(ID,contains("NA")) 


lettersGeno=read.table(opt$geno, col.names =colnames(genoHead),stringsAsFactors = F,colClasses = c("character")) %>% select(REF, ALT)

refAllele=lettersGeno$REF
altAllele=lettersGeno$ALT


genoMelt=melt(geno, id.vars = "ID", value.name = "FullGeno", variable.name = "Individual" ) %>% separate(FullGeno, into=c("geno","dose","extra1"), sep=":") %>% select(Individual, dose) %>% mutate(genotype=ifelse(round(as.integer(dose))==0, paste(refAllele, refAllele, sep=""), ifelse(round(as.integer(dose))==1, paste(refAllele,altAllele, sep=""), paste(altAllele,altAllele,sep=""))))
genoMelt$Individual= as.character(genoMelt$Individual)


pheno_qtlpeak=meltpheno %>% inner_join(genoMelt, by="Individual") %>%  mutate(PAU=num/denom) 

qtlplot=ggplot(pheno_qtlpeak, aes(x=genotype, y=PAU, fill=genotype)) + geom_boxplot(width=.5)+ geom_jitter(alpha=1) + facet_grid(~PeakLoc) +scale_fill_brewer(palette = "YlOrRd")

ggsave(plot=qtlplot, filename=opt$output, height=10, width=10)
```



Code for boxplots:  

run_qtlFacetBoxplots.sh
```{bash,eval=F}
#!/bin/bash

#SBATCH --job-name=qtlFacetBoxplots
#SBATCH --account=pi-yangili1
#SBATCH --time=24:00:00
#SBATCH --output=qtlFacetBoxplots.out
#SBATCH --error=qtlFacetBoxplots.err
#SBATCH --partition=broadwl
#SBATCH --mem=18G
#SBATCH --mail-type=END


module load Anaconda3
source activate three-prime-env

Fraction=$1
gene=$2
chrom=$3
snp=$4
peakID=$5


less /project2/gilad/briana/apaQTL/data/phenotype_5perc/APApeak_Phenotype_GeneLocAnno.${Fraction}.5perc.fc.gz | grep ${gene}_ > /project2/gilad/briana/apaQTL/data/ExampleQTLPlots/${gene}_${Fraction}PeaksPheno.txt


less /project2/gilad/briana/YRI_geno_hg19/chr${chrom}.dose.filt.vcf.gz | grep ${snp} >  /project2/gilad/briana/apaQTL/data/ExampleQTLPlots/${gene}_${Fraction}PeaksGenotype.txt

Rscript apaqtlfacetboxplots.R -P /project2/gilad/briana/apaQTL/data/ExampleQTLPlots/${gene}_${Fraction}PeaksPheno.txt -G /project2/gilad/briana/apaQTL/data/ExampleQTLPlots/${gene}_${Fraction}PeaksGenotype.txt --gene ${gene} -p ${peakID}  -o /project2/gilad/briana/apaQTL/data/ExampleQTLPlots/${gene}_${Fraction}${SNP}${peakID}_boxplot.png


```

```{r}
totAPAinTot %>% filter(abs(Originalslope)>1)
```

```{bash,eval=F}
sbatch run_qtlFacetBoxplots.sh "Total" "NBPF9" "1" "1:144701300" "peak7314"
sbatch run_qtlFacetBoxplots.sh "Total" "HLA-DRB5" "6" "6:32486756" "peak113454"
sbatch run_qtlFacetBoxplots.sh "Total" "HLA-DRB6" "6" "6:32538598" "peak113461"

```


```{r}
nucAPAinNuc %>% filter(abs(Originalslope)>1)
```


```{bash,eval=F}
sbatch run_qtlFacetBoxplots.sh "Nuclear" "LINC00869" "1" "1:149598905" "peak7883"
sbatch run_qtlFacetBoxplots.sh "Nuclear" "FRG1BP" "20" "20:29641550" "peak80905"
sbatch run_qtlFacetBoxplots.sh "Nuclear" "HLA-DRB5" "6" "6:32468906" "peak113456"

#test case  
sbatch run_qtlFacetBoxplots.sh "Nuclear" "TAF3" "10" "10:7980931" "peak14035"

```
