---
title: "Premature Termination QTL"
author: "Briana Mittleman"
date: "7/1/2019"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
```{r}
library(reshape2)
library(workflowr)
library(tidyverse)
```

Many papers have started to talk about premature termination. Premature terminated isoforms may be truncated protein or may be degraded. I am going to create a measure for this and test for genetic variation associated with it in my data. The measure will be sum of the reads in intronic PAS and the sum of the UTR reads. I will use leafcutter to put the ratios onto a normal distribution. I will then test for QTLs these ratios.  

```{bash,eval=F}
mkdir ../data/PreTerm_pheno
```

##Prepare phenotype
###Total

gene start and end 

```{r}
genes=read.table("/project2/gilad/briana/genome_anotation_data/RefSeq_annotations/FullTranscriptByName.bed", col.names = c("chr", "Gene_start", "Gene_end", "gene", "score", "strand"),stringsAsFactors = F) %>% select(chr,Gene_start, Gene_end, gene)
```

```{r}
totalPAS=read.table("../data/phenotype_5perc/APApeak_Phenotype_GeneLocAnno.Total.5perc.fc.gz",stringsAsFactors = F,header = T) 


totalPASPheno=totalPAS %>% melt(id.vars="chrom", variable.name="Ind", value.name = "ratio") %>% separate(ratio, into=c("count", "geneCount"), sep="/") %>% separate(chrom, into=c("chr", "start", "end", "geneID"), sep=":") %>% separate(geneID, into=c("gene", "loc","strand", "PAS"), sep="_") %>% filter(loc=="utr3" | loc=="intron") %>% inner_join(genes,by=c("chr", "gene"))%>% mutate(gene=paste(chr,Gene_start, Gene_end, gene,sep=":")) %>% group_by(Ind,gene,loc) %>% summarise(SumCount=sum(as.integer(count))) %>% ungroup() %>% group_by(Ind,gene) %>% mutate(nType=n()) %>% filter(nType==2) %>% spread(loc, SumCount) %>% mutate(total=intron+utr3,PreTermInt=paste(intron,total, sep="/"),PreTermUTR=paste(utr3,total, sep="/")) %>% select(-nType, -intron,-utr3,-total)


totalPASPheno_melt= totalPASPheno %>% melt(id.vars=c("Ind", "gene"), variable.name="Type", value.name = "Value") %>% mutate(chrom=paste(gene, Type, sep="_")) %>% spread(Ind, Value) %>% select(-gene, -Type)


#write.table(totalPASPheno_melt,"../data/PreTerm_pheno/Total_preterminationPheno.txt",quote=F, row.names=F,col.names=T, sep=" ")

```



```{bash,eval=F}
#python2
gzip ../data/PreTerm_pheno/Total_preterminationPheno.txt
python prepare_phenotype_table.py ../data/PreTerm_pheno/Total_preterminationPheno.txt.gz

#activate env  

sh ../data/PreTerm_pheno/Total_preterminationPheno.txt.gz_prepare.sh

#top 2 pcs
head -n 3  ../data/PreTerm_pheno/Total_preterminationPheno.txt.gz.PCs > ../data/PreTerm_pheno/Total_preterminationPheno.txt.gz.2PCs 
```

###Nuclear  


```{r}
nuclearPAS=read.table("../data/phenotype_5perc/APApeak_Phenotype_GeneLocAnno.Nuclear.5perc.fc.gz",stringsAsFactors = F,header = T) 


nuclearPASPheno=nuclearPAS %>% melt(id.vars="chrom", variable.name="Ind", value.name = "ratio") %>% separate(ratio, into=c("count", "geneCount"), sep="/") %>% separate(chrom, into=c("chr", "start", "end", "geneID"), sep=":") %>% separate(geneID, into=c("gene", "loc","strand", "PAS"), sep="_") %>% filter(loc=="utr3" | loc=="intron") %>% inner_join(genes,by=c("chr", "gene"))%>% mutate(gene=paste(chr,Gene_start, Gene_end, gene,sep=":")) %>% group_by(Ind,gene,loc) %>% summarise(SumCount=sum(as.integer(count))) %>% ungroup() %>% group_by(Ind,gene) %>% mutate(nType=n()) %>% filter(nType==2) %>% spread(loc, SumCount) %>% mutate(total=intron+utr3,PreTermInt=paste(intron,total, sep="/"),PreTermUTR=paste(utr3,total, sep="/")) %>% select(-nType, -intron,-utr3,-total)


nuclearPASPheno_melt= nuclearPASPheno %>% melt(id.vars=c("Ind", "gene"), variable.name="Type", value.name = "Value") %>% mutate(chrom=paste(gene, Type, sep="_")) %>% spread(Ind, Value) %>% select(-gene, -Type)


#write.table(nuclearPASPheno_melt,"../data/PreTerm_pheno/Nuclear_preterminationPheno.txt",quote=F, row.names=F,col.names=T, sep=" ")

```


```{bash,eval=F}
#python2
gzip ../data/PreTerm_pheno/Nuclear_preterminationPheno.txt
python prepare_phenotype_table.py ../data/PreTerm_pheno/Nuclear_preterminationPheno.txt.gz
#env
sh ../data/PreTerm_pheno/Nuclear_preterminationPheno.txt.gz_prepare.sh

#top 2 pcs
head -n 3  ../data/PreTerm_pheno/Nuclear_preterminationPheno.txt.gz.PCs > ../data/PreTerm_pheno/Nuclear_preterminationPheno.txt.gz.2PCs 
```

##Call QTLs

Sample list from previous work

```{bash,eval=F}
mkdir ../data/PrematureQTLNominal
mkdir ../data/PrematureQTLPermuted
```


```{bash,eval=F}
sbatch PrematureQTLNominal.sh
sbatch PrematureQTLPermuted.sh
```

May want to only test one number per gene but do this for now because I want to take advantage of the leafcutter normalization software.  


```{bash,eval=F}
cat ../data/PrematureQTLPermuted/Total_preterminationPheno.txt.gz.qqnorm_chr* > ../data/PrematureQTLPermuted/Total_preterminationPheno.txt.gz.qqnorm_AllChr.txt

cat ../data/PrematureQTLPermuted/Nuclear_preterminationPheno.txt.gz.qqnorm_chr* > ../data/PrematureQTLPermuted/Nuclear_preterminationPheno.txt.gz.qqnorm_AllChr.txt
```
####Tot

```{r}
totRes=read.table("../data/PrematureQTLPermuted/Total_preterminationPheno.txt.gz.qqnorm_AllChr.txt", stringsAsFactors = F,col.names = c("pid", "nvar", "shape1", "shape2", "dummy", "sid", "dist", "npval", "slope", "ppval", "bpval"))

totRes$bh=p.adjust(totRes$bpval, method="fdr")

totRes_sig=totRes %>% filter(-log10(bh)>1) 


totRes_sig_genes=totRes_sig %>% separate(pid, into=c("chr","start","end", "geneID"), sep=":") %>% separate(geneID, into=c("gene", "Frac"),sep="_") %>% select(gene) %>% unique()

write.table(totRes, file = "../data/PrematureQTLPermuted/Total_preterminationPheno.txt.gz.qqnorm_AllChrBH.txt", col.names = T, row.names = F, quote = F)
```

```{r}
nrow(totRes_sig_genes)
```


Proportion of genes tested:

```{r}
tottested_genes=totRes %>% separate(pid, into=c("chr","start","end", "geneID"), sep=":") %>% separate(geneID, into=c("gene", "Frac"),sep="_") %>% select(gene) %>% unique()
nrow(totRes_sig_genes)/nrow(tottested_genes)
```
qqplot: 

```{r}
qqplot(-log10(runif(nrow(totRes))), -log10(totRes$bpval),ylab="-log10 Total permuted pvalue", xlab="Uniform expectation", main="Total premature termination")
abline(0,1)
```

####Nuclear:

```{r}
nucRes=read.table("../data/PrematureQTLPermuted/Nuclear_preterminationPheno.txt.gz.qqnorm_AllChr.txt", stringsAsFactors = F,col.names = c("pid", "nvar", "shape1", "shape2", "dummy", "sid", "dist", "npval", "slope", "ppval", "bpval"))

nucRes$bh=p.adjust(nucRes$bpval, method="fdr")

nucRes_sig=nucRes %>% filter(-log10(bh)>1)


nucRes_sig_genes=nucRes_sig %>% separate(pid, into=c("chr","start","end", "geneID"), sep=":") %>% separate(geneID, into=c("gene", "Frac"),sep="_") %>% select(gene) %>% unique()


write.table(nucRes, file = "../data/PrematureQTLPermuted/Nuclear_preterminationPheno.txt.gz.qqnorm_AllChrBH.txt", col.names = T, row.names = F, quote = F)
```

```{r}
nrow(nucRes_sig_genes)
```
Proportion of genes tested:

```{r}
nuctested_genes=nucRes %>% separate(pid, into=c("chr","start","end", "geneID"), sep=":") %>% separate(geneID, into=c("gene", "Frac"),sep="_") %>% select(gene) %>% unique()
nrow(nucRes_sig_genes)/nrow(nuctested_genes)
```


qqplot: 

```{r}
qqplot(-log10(runif(nrow(nucRes))), -log10(nucRes$bpval),ylab="-log10 Nuclear permuted pvalue", xlab="Uniform expectation", main="Nuclear premature termination")
abline(0,1)
```
More likely in nuclear:

```{r}

prop.test(x=c(nrow(nucRes_sig_genes),nrow(totRes_sig_genes)), n=c(nrow(nuctested_genes),nrow(tottested_genes)),alternative = "greater")
```
###overlap with eGenes  
I next want to look at the proportion of eGenes. 


```{r}
explainedEgenes=read.table("../data/Li_eQTLs/explainedEgenes.txt", col.names = c("gene"), stringsAsFactors = F)
unexplainedEgenes=read.table("../data/Li_eQTLs/UnexplainedEgenes.txt", col.names = c("gene"), stringsAsFactors = F)


allEgenes=bind_rows(explainedEgenes, unexplainedEgenes)

```

I want to test the proportion of overlap.

```{r}
TotPre_uneGene=totRes_sig_genes %>% inner_join(unexplainedEgenes,by="gene")
NucPre_uneGene=nucRes_sig_genes %>% inner_join(unexplainedEgenes,by="gene")

TotPre_exeGene=totRes_sig_genes %>% inner_join(explainedEgenes,by="gene")
NucPre_exeGene=nucRes_sig_genes %>% inner_join(explainedEgenes,by="gene")

TotPre_alleGene=totRes_sig_genes %>% inner_join(allEgenes,by="gene")
NucPre_alleGene=nucRes_sig_genes %>% inner_join(allEgenes,by="gene")
```


Proportion of eGenes explaiend by this:

```{r}
#total

nrow(TotPre_uneGene)/nrow(unexplainedEgenes)
nrow(TotPre_exeGene)/nrow(explainedEgenes)
nrow(TotPre_alleGene)/nrow(allEgenes)
```
```{r}
#nuclear

nrow(NucPre_uneGene)/nrow(unexplainedEgenes)
nrow(NucPre_exeGene)/nrow(explainedEgenes)
nrow(NucPre_alleGene)/nrow(allEgenes)
```


```{r}
prop.test(x=c(nrow(NucPre_uneGene),nrow(TotPre_uneGene)), n=c(nrow(unexplainedEgenes),nrow(unexplainedEgenes)))


prop.test(x=c(nrow(NucPre_exeGene),nrow(TotPre_exeGene)), n=c(nrow(explainedEgenes),nrow(explainedEgenes)))
```
Conclusion: 

Total- 13 overlaps with all eGenes, 7 ex, 6 unexplained
Nuclear- 24 overlaps with all eGenes, 13 ex, 11 unexpained 

All eGenes=1824
Unexplained=760
Explained=1064
